<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Introduction to NCBI databases | Computational Biology for All!</title>
  <meta name="description" content="A bioinformatics and computational biology book" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Introduction to NCBI databases | Computational Biology for All!" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A bioinformatics and computational biology book" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Introduction to NCBI databases | Computational Biology for All!" />
  
  <meta name="twitter:description" content="A bioinformatics and computational biology book" />
  

<meta name="author" content="Avril Coglan and Nathan Brouwer" />


<meta name="date" content="2021-01-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="downloading-ncbi-sequence-data-by-hand.html"/>
<link rel="next" href="downloading-sequences-from-uniprot-by-hand.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Computational Biology for All</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="downloading-r.html"><a href="downloading-r.html"><i class="fa fa-check"></i><b>2</b> Downloading R</a><ul>
<li class="chapter" data-level="2.1" data-path="downloading-r.html"><a href="downloading-r.html#introduction-to-r"><i class="fa fa-check"></i><b>2.1</b> Introduction to R</a></li>
<li class="chapter" data-level="2.2" data-path="downloading-r.html"><a href="downloading-r.html#installing-r"><i class="fa fa-check"></i><b>2.2</b> Installing R</a><ul>
<li class="chapter" data-level="2.2.1" data-path="downloading-r.html"><a href="downloading-r.html#installing-r-on-a-windows-pc"><i class="fa fa-check"></i><b>2.2.1</b> Installing R on a Windows PC</a></li>
<li class="chapter" data-level="2.2.2" data-path="downloading-r.html"><a href="downloading-r.html#how-to-install-r-on-non-windows-computers-eg.-macintosh-or-linux-computers"><i class="fa fa-check"></i><b>2.2.2</b> How to install R on non-Windows computers (eg. Macintosh or Linux computers)</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="downloading-r.html"><a href="downloading-r.html#starting-r"><i class="fa fa-check"></i><b>2.3</b> Starting R</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="downloading-the-rstudio.html"><a href="downloading-the-rstudio.html"><i class="fa fa-check"></i><b>3</b> Downloading the RStudio</a><ul>
<li class="chapter" data-level="3.1" data-path="downloading-the-rstudio.html"><a href="downloading-the-rstudio.html#installing-the-rstudio-ide"><i class="fa fa-check"></i><b>3.1</b> Installing the RStudio IDE</a></li>
<li class="chapter" data-level="3.2" data-path="downloading-the-rstudio.html"><a href="downloading-the-rstudio.html#getting-to-know-rstudio"><i class="fa fa-check"></i><b>3.2</b> Getting to know RStudio</a></li>
<li class="chapter" data-level="3.3" data-path="downloading-the-rstudio.html"><a href="downloading-the-rstudio.html#rstudio-versus-rstudio-cloud"><i class="fa fa-check"></i><b>3.3</b> RStudio versus RStudio Cloud</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="installing-r-packages.html"><a href="installing-r-packages.html"><i class="fa fa-check"></i><b>4</b> Installing R packages</a><ul>
<li class="chapter" data-level="4.1" data-path="installing-r-packages.html"><a href="installing-r-packages.html#downloading-packages-with-the-rstudio-ide"><i class="fa fa-check"></i><b>4.1</b> Downloading packages with the RStudio IDE</a></li>
<li class="chapter" data-level="4.2" data-path="installing-r-packages.html"><a href="installing-r-packages.html#downloading-packages-with-the-function-install.packages"><i class="fa fa-check"></i><b>4.2</b> Downloading packages with the function install.packages()</a></li>
<li class="chapter" data-level="4.3" data-path="installing-r-packages.html"><a href="installing-r-packages.html#using-packages-after-they-are-downloaded"><i class="fa fa-check"></i><b>4.3</b> Using packages after they are downloaded</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html"><i class="fa fa-check"></i><b>5</b> Installing Bioconductor</a><ul>
<li class="chapter" data-level="5.1" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html#bioconductor"><i class="fa fa-check"></i><b>5.1</b> Bioconductor</a></li>
<li class="chapter" data-level="5.2" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html#installing-biocmanager"><i class="fa fa-check"></i><b>5.2</b> Installing BiocManager</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html"><i class="fa fa-check"></i><b>6</b> Brief introduction to R</a><ul>
<li class="chapter" data-level="6.1" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#vocabulary"><i class="fa fa-check"></i><b>6.1</b> Vocabulary</a></li>
<li class="chapter" data-level="6.2" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#r-functions-etc"><i class="fa fa-check"></i><b>6.2</b> R functions etc</a></li>
<li class="chapter" data-level="6.3" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#a-brief-introduction-to-r"><i class="fa fa-check"></i><b>6.3</b> A brief introduction to R</a></li>
<li class="chapter" data-level="6.4" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#links-and-further-reading"><i class="fa fa-check"></i><b>6.4</b> Links and Further Reading</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="downloading-ncbi-sequence-data-by-hand.html"><a href="downloading-ncbi-sequence-data-by-hand.html"><i class="fa fa-check"></i><b>7</b> Downloading NCBI sequence data by hand</a><ul>
<li class="chapter" data-level="7.1" data-path="downloading-ncbi-sequence-data-by-hand.html"><a href="downloading-ncbi-sequence-data-by-hand.html#retrieving-genome-sequence-data-via-the-ncbi-website"><i class="fa fa-check"></i><b>7.1</b> Retrieving genome sequence data via the NCBI website</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html"><i class="fa fa-check"></i><b>8</b> Introduction to NCBI databases</a><ul>
<li class="chapter" data-level="8.1" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#to-do"><i class="fa fa-check"></i><b>8.1</b> To do</a></li>
<li class="chapter" data-level="8.2" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#the-ncbi-sequence-database"><i class="fa fa-check"></i><b>8.2</b> The NCBI Sequence Database</a></li>
<li class="chapter" data-level="8.3" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#the-ncbi-sub-databases"><i class="fa fa-check"></i><b>8.3</b> The NCBI Sub-Databases</a></li>
<li class="chapter" data-level="8.4" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#searching-for-an-accession-number-in-the-ncbi-database"><i class="fa fa-check"></i><b>8.4</b> Searching for an accession number in the NCBI database</a></li>
<li class="chapter" data-level="8.5" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#ncbi-sequence-format-ncbi-format"><i class="fa fa-check"></i><b>8.5</b> NCBI Sequence Format (NCBI Format)</a></li>
<li class="chapter" data-level="8.6" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#refseq"><i class="fa fa-check"></i><b>8.6</b> RefSeq</a></li>
<li class="chapter" data-level="8.7" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#querying-the-ncbi-database"><i class="fa fa-check"></i><b>8.7</b> Querying the NCBI Database</a></li>
<li class="chapter" data-level="8.8" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#querying-the-ncbi-database-via-the-ncbi-website-for-reference"><i class="fa fa-check"></i><b>8.8</b> Querying the NCBI Database via the NCBI Website (for reference)</a></li>
<li class="chapter" data-level="8.9" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#example-finding-the-sequences-published-in-nature-460352-358-for-reference"><i class="fa fa-check"></i><b>8.9</b> Example: finding the sequences published in Nature 460:352-358 (for reference)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html"><i class="fa fa-check"></i><b>9</b> Downloading sequences from UniProt by hand</a><ul>
<li class="chapter" data-level="9.1" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#vocab"><i class="fa fa-check"></i><b>9.1</b> Vocab</a></li>
<li class="chapter" data-level="9.2" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#uniprot"><i class="fa fa-check"></i><b>9.2</b> UniProt</a></li>
<li class="chapter" data-level="9.3" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#viewing-the-uniprot-webpage-for-a-protein-sequence"><i class="fa fa-check"></i><b>9.3</b> Viewing the UniProt webpage for a protein sequence</a><ul>
<li class="chapter" data-level="9.3.1" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#protein-function"><i class="fa fa-check"></i><b>9.3.1</b> Protein function</a></li>
<li class="chapter" data-level="9.3.2" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#protein-sequence-and-size"><i class="fa fa-check"></i><b>9.3.2</b> Protein sequence and size</a></li>
<li class="chapter" data-level="9.3.3" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#other-information"><i class="fa fa-check"></i><b>9.3.3</b> Other information</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#retrieving-a-uniprot-protein-sequence-via-the-uniprot-website"><i class="fa fa-check"></i><b>9.4</b> Retrieving a UniProt protein sequence via the UniProt website</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="sequence-databases.html"><a href="sequence-databases.html"><i class="fa fa-check"></i><b>10</b> Sequence Databases</a><ul>
<li class="chapter" data-level="10.1" data-path="sequence-databases.html"><a href="sequence-databases.html#querying-the-ncbi-database-via-r"><i class="fa fa-check"></i><b>10.1</b> Querying the NCBI Database via R</a></li>
<li class="chapter" data-level="10.2" data-path="sequence-databases.html"><a href="sequence-databases.html#summary"><i class="fa fa-check"></i><b>10.2</b> Summary</a></li>
<li class="chapter" data-level="10.3" data-path="sequence-databases.html"><a href="sequence-databases.html#acknowledgements"><i class="fa fa-check"></i><b>10.3</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html"><i class="fa fa-check"></i><b>11</b> Local variation in GC content - NOT UPDATED</a><ul>
<li class="chapter" data-level="11.0.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#note-on-the-biology-in-this-section"><i class="fa fa-check"></i><b>11.0.1</b> Note on the biology in this section</a></li>
<li class="chapter" data-level="11.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#vocabulary-1"><i class="fa fa-check"></i><b>11.1</b> Vocabulary</a></li>
<li class="chapter" data-level="11.2" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#reading-sequence-data-with-rentrezentrez_fetch"><i class="fa fa-check"></i><b>11.2</b> Reading sequence data with rentrez::entrez_fetch</a></li>
<li class="chapter" data-level="11.3" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#local-variation-in-gc-content"><i class="fa fa-check"></i><b>11.3</b> Local variation in GC content</a></li>
<li class="chapter" data-level="11.4" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#a-sliding-window-analysis-of-gc-content"><i class="fa fa-check"></i><b>11.4</b> A sliding window analysis of GC content</a></li>
<li class="chapter" data-level="11.5" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#acknowledgements-1"><i class="fa fa-check"></i><b>11.5</b> Acknowledgements</a><ul>
<li class="chapter" data-level="11.5.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#license"><i class="fa fa-check"></i><b>11.5.1</b> License</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#exercises"><i class="fa fa-check"></i><b>11.6</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html"><i class="fa fa-check"></i><b>12</b> Alignments in R - NOT UPDAYED</a><ul>
<li class="chapter" data-level="12.1" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#retrieving-a-uniprot-protein-sequence-using-seqinr"><i class="fa fa-check"></i><b>12.1</b> Retrieving a UniProt protein sequence using SeqinR</a></li>
<li class="chapter" data-level="12.2" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#comparing-two-sequences-using-a-dotplot"><i class="fa fa-check"></i><b>12.2</b> Comparing two sequences using a dotplot</a></li>
<li class="chapter" data-level="12.3" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#aligning-uniprot-sequences"><i class="fa fa-check"></i><b>12.3</b> Aligning UniProt sequences</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html"><i class="fa fa-check"></i><b>13</b> Multiple alignment with Clustal</a><ul>
<li class="chapter" data-level="13.1" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#installing-the-clustal-multiple-alignment-software"><i class="fa fa-check"></i><b>13.1</b> Installing the CLUSTAL multiple alignment software</a></li>
<li class="chapter" data-level="13.2" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#discarding-very-poorly-conserved-regions-from-an-alignment"><i class="fa fa-check"></i><b>13.2</b> Discarding very poorly conserved regions from an alignment</a></li>
<li class="chapter" data-level="13.3" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#calculating-genetic-distances-between-protein-sequences"><i class="fa fa-check"></i><b>13.3</b> Calculating genetic distances between protein sequences</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html"><i class="fa fa-check"></i><b>14</b> Unrooted Neighbor-joining Phylogenetic trees</a><ul>
<li class="chapter" data-level="14.1" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#building-an-unrooted-phylogenetic-tree-for-protein-sequences"><i class="fa fa-check"></i><b>14.1</b> Building an unrooted phylogenetic tree for protein sequences</a></li>
<li class="chapter" data-level="14.2" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#testing-the-significance-of-an-alignment"><i class="fa fa-check"></i><b>14.2</b> Testing the significance of an alignment</a></li>
<li class="chapter" data-level="14.3" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#calculating-the-statistical-significance-of-a-pairwise-global-alignment"><i class="fa fa-check"></i><b>14.3</b> Calculating the statistical significance of a pairwise global alignment</a></li>
<li class="chapter" data-level="14.4" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#summary-1"><i class="fa fa-check"></i><b>14.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html"><i class="fa fa-check"></i><b>15</b> Retrieving multiple sequences in R</a><ul>
<li class="chapter" data-level="15.1" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html#prelminaries"><i class="fa fa-check"></i><b>15.1</b> Prelminaries</a></li>
<li class="chapter" data-level="15.2" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html#retrieving-a-list-of-sequences-from-uniprot"><i class="fa fa-check"></i><b>15.2</b> Retrieving a list of sequences from UniProt</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html"><i class="fa fa-check"></i><b>16</b> Computational gene finding</a><ul>
<li class="chapter" data-level="16.1" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#the-genetic-code"><i class="fa fa-check"></i><b>16.1</b> The genetic code</a></li>
<li class="chapter" data-level="16.2" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#reading-frames"><i class="fa fa-check"></i><b>16.2</b> Reading frames</a></li>
<li class="chapter" data-level="16.3" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#finding-open-reading-frames-on-the-forward-strand-of-a-dna-sequence"><i class="fa fa-check"></i><b>16.3</b> Finding open reading frames on the forward strand of a DNA sequence</a></li>
<li class="chapter" data-level="16.4" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#lengths-of-open-reading-frames"><i class="fa fa-check"></i><b>16.4</b> Lengths of open reading frames</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html"><i class="fa fa-check"></i><b>17</b> Significance of ORFs</a><ul>
<li class="chapter" data-level="17.1" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html#identifying-significant-open-reading-frames"><i class="fa fa-check"></i><b>17.1</b> Identifying significant open reading frames</a></li>
<li class="chapter" data-level="17.2" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html#summary-2"><i class="fa fa-check"></i><b>17.2</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="rooted-neighbor-joining-phylogenetic-trees.html"><a href="rooted-neighbor-joining-phylogenetic-trees.html"><i class="fa fa-check"></i><b>18</b> Rooted Neighbor-joining Phylogenetic trees</a><ul>
<li class="chapter" data-level="18.1" data-path="rooted-neighbor-joining-phylogenetic-trees.html"><a href="rooted-neighbor-joining-phylogenetic-trees.html#building-a-rooted-phylogenetic-tree-for-protein-sequences"><i class="fa fa-check"></i><b>18.1</b> Building a rooted phylogenetic tree for protein sequences</a></li>
<li class="chapter" data-level="18.2" data-path="rooted-neighbor-joining-phylogenetic-trees.html"><a href="rooted-neighbor-joining-phylogenetic-trees.html#building-a-phylogenetic-tree-for-dna-or-mrna-sequences"><i class="fa fa-check"></i><b>18.2</b> Building a phylogenetic tree for DNA or mRNA sequences</a></li>
<li class="chapter" data-level="18.3" data-path="rooted-neighbor-joining-phylogenetic-trees.html"><a href="rooted-neighbor-joining-phylogenetic-trees.html#summary-3"><i class="fa fa-check"></i><b>18.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="todo.html"><a href="todo.html"><i class="fa fa-check"></i><b>19</b> TODO</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://https://github.com/brouwern/CB4ALL" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Computational Biology for All!</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introduction-to-ncbi-databases" class="section level1">
<h1><span class="header-section-number">Chapter 8</span> Introduction to NCBI databases</h1>
<p><strong>By</strong>: Avril Coghlan. Adapted and edited by Nathan Brouwer.</p>
<p>This vignette provides an introduction to the general search features of the NCBI databases, including how to search by both accession numbers and other search parameters, such as specific papers. For example, if you don’t know the accession number of a sequence, you can locate it via a paper that worked on that sequence.</p>
<div id="to-do" class="section level2">
<h2><span class="header-section-number">8.1</span> To do</h2>
<ul>
<li>replicate searches in R using rentrez</li>
</ul>
</div>
<div id="the-ncbi-sequence-database" class="section level2">
<h2><span class="header-section-number">8.2</span> The NCBI Sequence Database</h2>
<p>All published genome sequences are available over the internet, as it is a requirement of every scientific journal that any published DNA or RNA or protein sequence must be deposited in a public database. The main resources for storing and distributing sequence data are three large databases: the <strong>NCBI database</strong> (www.ncbi.nlm.nih.gov/), the *European Molecular Biology Laboratory (EMBL)<strong> database (<a href="https://www.ebi.ac.uk/ena" class="uri">https://www.ebi.ac.uk/ena</a>), and the </strong>DNA Database of Japan (DDBJ)** database (www.ddbj.nig.ac.jp/). These databases collect all publicly available DNA, RNA and protein sequence data and make it available for free. They exchange data nightly, so contain essentially the same data.</p>
<p>In this vignette we will discuss the <strong>NCBI database</strong>. Note however that it contains essentially the same data as in the EMBL/DDBJ databases.</p>
<p>Sequences in the NCBI Sequence Database are identified by an <strong>accession number</strong>. This is a unique number that is <em>only</em> associated with one sequence. For example, the accession number NC_001477 is for the DEN-1 Dengue virus genome sequence. The accession number is what identifies the sequence. It is reported in scientific papers describing that sequence.</p>
<p>As well as the sequence itself, for each sequence the NCBI database also stores some additional <strong>annotation</strong> data, such as the name of the species it comes from, references to publications describing that sequence, information on the structure of the protein, etc. Some of this annotation data was added by the person who sequenced a sequence and submitted it to the NCBI database, while some may have been added later by a human curator working for NCBI.</p>
</div>
<div id="the-ncbi-sub-databases" class="section level2">
<h2><span class="header-section-number">8.3</span> The NCBI Sub-Databases</h2>
<p>The NCBI database contains several sub-databases, the most important of which are:</p>
<ul>
<li>Nucleotide database: contains DNA and RNA sequences</li>
<li>NCBI Protein database: contains protein sequences</li>
<li>EST database: contains ESTs (expressed sequence tags), which are short sequences derived from mRNAs</li>
<li>Genome database: contains DNA sequences for whole genomes</li>
<li>PubMed: contains data on scientific publications</li>
</ul>
</div>
<div id="searching-for-an-accession-number-in-the-ncbi-database" class="section level2">
<h2><span class="header-section-number">8.4</span> Searching for an accession number in the NCBI database</h2>
<p>The <strong>FASTA</strong> file format is a file format commonly used to store sequence information. The first line starts with the greater than character followed by a name and/or description for the sequence. Subsequent lines contain the sequence itself.</p>
<pre><code>## &gt;mysequence1
## ACATGAGACAGACAGACCCCCAGAGACAGACCCCTAGACACAGAGAGAG
## TATGCAGGACAGGGTTTTTGCCCAGGGTGGCAGTATG</code></pre>
<p>A FASTA file can contain more than one sequence. If a FASTA file contains many sequences, then for each sequence it will have a <strong>header line</strong> starting with greater than character followed by the sequence itself.</p>
<pre><code>## &gt;mysequence1
## ACATGAGACAGACAGACCCCCAGAGACAGACCCCTAGACACAGAGAGAG
## TATGCAGGACAGGGTTTTTGCCCAGGGTGGCAGTATG
## 
## &gt;mysequence2
## AGGATTGAGGTATGGGTATGTTCCCGATTGAGTAGCCAGTATGAGCCAG
## AGTTTTTTACAAGTATTTTTCCCAGTAGCCAGAGAGAGAGTCACCCAGT
## ACAGAGAGC</code></pre>
</div>
<div id="ncbi-sequence-format-ncbi-format" class="section level2">
<h2><span class="header-section-number">8.5</span> NCBI Sequence Format (NCBI Format)</h2>
<p>As mentioned above, for each sequence the NCBI database stores some extra information such as the species that it came from, publications describing the sequence, etc. This information is stored in the NCBI entry or NCBI record for the sequence. The NCBI entry for a sequence can be viewed by searching the NCBI database for the accession number for that sequence. The NCBI entries for sequences are stored in a particular format, known as <strong>NCBI format</strong>.</p>
<p>To view the NCBI entry for the DEN-1 Dengue virus (which has accession NC_001477), follow these steps:</p>
<ol style="list-style-type: decimal">
<li>Go to the NCBI website (www.ncbi.nlm.nih.gov).</li>
<li>Search for the accession number.</li>
<li>On the results page, if your sequence corresponds to a nucleotide (DNA or RNA) sequence, you should see a hit in the Nucleotide database, and you should click on the word “Nucleotide” to view the NCBI entry for the hit. Likewise, if your sequence corresponds to a protein sequence, you should see a hit in the Protein database, and you should click on the word “Protein” to view the NCBI entry for the hit.</li>
<li>After you click on “Nucleotide” or “Protein” in the previous step, the NCBI entry for the accession will appear.</li>
</ol>
<p>The NCBI entry for an accession contains a lot of information about the sequence, such as papers describing it, features in the sequence, etc. The <strong>DEFINITION</strong> field gives a short description for the sequence. The <strong>ORGANISM</strong> field in the NCBI entry identifies the species that the sequence came from. The <strong>REFERENCE</strong> field contains scientific publications describing the sequence. The <strong>FEATURES</strong> field contains information about the location of features of interest inside the sequence, such as regulatory sequences or genes that lie inside the sequence. The <strong>ORIGIN</strong> field gives the sequence itself.</p>
</div>
<div id="refseq" class="section level2">
<h2><span class="header-section-number">8.6</span> RefSeq</h2>
<p>When carrying out searches of the NCBI database, it is important to bear in mind that the database may contain <strong>redundant sequences</strong> for the same gene that were sequenced by different laboratories (because many different labs have sequenced the gene, and submitted their sequences to the NCBI database).</p>
<p>There are also many different types of nucleotide sequences and protein sequences in the NCBI database. With respect to nucleotide sequences, some many be entire genomic DNA sequences, some may be mRNAs, and some may be lower quality sequences such as expressed sequence tags (ESTs, which are derived from parts of mRNAs), or DNA sequences of contigs from genome projects. That is, you can end up with an entry in the protein database based on sequence derived from a genomic sequence, from sequencing just the gene, and from other routes.</p>
<p>Furthermore, some sequences may be <strong>manually curated</strong> so that the associated entries contain extra information, but the majority of sequences are <strong>uncurated.</strong></p>
<p>As mentioned above, the NCBI database often contains redundant information for a gene, contains sequences of varying quality, and contains both uncurated and curated data.</p>
<p>As a result, NCBI has made a special database called <strong>RefSeq (reference sequence database)</strong>, which is a subset of the NCBI database. The data in RefSeq is manually curated, is high quality sequence data, and is non-redundant; this means that each gene (or <strong>splice-form / isoform</strong> of a gene, in the case of eukaryotes), protein, or genome sequence is only represented once.</p>
<p>The data in RefSeq is curated and is of much higher quality than the rest of the NCBI Sequence Database. However, unfortunately, because of the high level of manual curation required, RefSeq does not cover all species, and is not comprehensive for the species that are covered so far. To speed up searches and simplify the results in to can be very useful to just search RefSeq. However, for detailed and thorough work the full database should probably be searched and the results scrutinized.</p>
<p>You can easily tell that a sequence comes from RefSeq because its accession number starts with particular sequence of letters. That is, accessions of RefSeq sequences corresponding to protein records usually start with <strong>NP_</strong>, and accessions of RefSeq curated complete genome sequences usually start with <strong>NC_</strong> or <strong>NS_</strong>.</p>
</div>
<div id="querying-the-ncbi-database" class="section level2">
<h2><span class="header-section-number">8.7</span> Querying the NCBI Database</h2>
<p>You may need to interrogate the NCBI Database to find particular sequences or a set of sequences matching given criteria, such as:</p>
<ul>
<li>The sequence with accession NC_001477</li>
<li>The sequences published in Nature 460:352-358</li>
<li>All sequences from <em>Chlamydia trachomatis</em></li>
<li>Sequences submitted by Caroline Cameron, a syphyllis researcher</li>
<li>Flagellin or fibrinogen sequences</li>
<li>The glutamine synthetase gene from <em>Mycobacteriuma leprae</em></li>
<li>Just the upstream control region of the <em>Mycobacterium leprae</em> dnaA gene</li>
<li>The sequence of the <em>Mycobacterium leprae</em> DnaA protein</li>
<li>The genome sequence of syphilis, <em>Treponema pallidum</em> subspp. <em>pallidum</em></li>
<li>All human nucleotide sequences associated with malaria</li>
</ul>
<p>There are two main ways that you can query the NCBI database to find these sets of sequences. The first possibility is to carry out searches on the NCBI website. The second possibility is to carry out searches from R using one of several packages that can interface with NCBI. As of October 2019 rentrez seems to be the best package for this..</p>
<p>Below, I will explain how to manually carry out queries on the NCBI database.</p>
</div>
<div id="querying-the-ncbi-database-via-the-ncbi-website-for-reference" class="section level2">
<h2><span class="header-section-number">8.8</span> Querying the NCBI Database via the NCBI Website (for reference)</h2>
<p><strong>NOTE</strong>: The following section is here for reference; you need to know its possible to refine searches but do not need to know any of these actual tags.</p>
<p>If you are carrying out searches on the NCBI website, to narrow down your searches to specific types of sequences or to specific organisms, you will need to use <strong>“search tags”.</strong></p>
<p>For example, the search tags “[PROP]” and “[ORGN]” let you restrict your search to a specific subset of the NCBI Sequence Database, or to sequences from a particular taxon, respectively. Here is a list of useful search tags, which we will explain how to use below:</p>
<ul>
<li>[AC] NC_001477[AC] With a particular accession number</li>
<li>[ORGN] Fungi[ORGN] From a particular organism or taxon</li>
<li>[PROP] biomol_mRNA[PROP] Of a specific type (eg. mRNA) or from a specific database (eg. RefSeq)</li>
<li>[JOUR] Nature[JOUR] Described in a paper published in a particular journal</li>
<li>[VOL] 531[VOL] Described in a paper published in a particular journal volume</li>
<li>[PAGE] 27[PAGE] Described in a paper with a particular start-page in a journal</li>
<li>[AU] “Smith J”[AU] Described in a paper, or submitted to NCBI, by a particular author</li>
</ul>
<p>To carry out searches of the NCBI database, you first need to go to the NCBI website, and type your search query into the search box at the top. For example, to search for all sequences from Fungi, you would type “Fungi[ORGN]” into the search box on the NCBI website.</p>
<p>You can combine the search tags above by using “AND”, to make more complex searches. For example, to find all mRNA sequences from Fungi, you could type “Fungi[ORGN] AND biomol_mRNA[PROP]” in the search box on the NCBI website.</p>
<p>Likewise, you can also combine search tags by using “OR”, for example, to search for all mRNA sequences from Fungi or Bacteria, you would type “(Fungi[ORGN] OR Bacteria[ORGN]) AND biomol_mRNA[PROP]” in the search box. Note that you need to put brackets around “Fungi[ORGN] OR Bacteria[ORGN]” to specify that the word “OR” refers to these two search tags.</p>
<p>Here are some examples of searches, some of them made by combining search terms using “AND”:</p>
<ul>
<li>NC_001477[AC] - With accession number NC_001477</li>
<li>Nature[JOUR] AND 460[VOL] AND 352[PAGE] - Published in Nature 460:352-358</li>
<li>“Chlamydia trachomatis”[ORGN] - From the bacterium Chlamydia trachomatis</li>
<li>“Berriman M”[AU] - Published in a paper, or submitted to NCBI, by M. Berriman</li>
<li>flagellin OR fibrinogen - Which contain the word “flagellin” or “fibrinogen” in their NCBI record</li>
<li>“Mycobacterium leprae”[ORGN] AND dnaA - Which are from M. leprae, and contain “dnaA” in their NCBI record</li>
<li>“Homo sapiens”[ORGN] AND “colon cancer” - Which are from human, and contain “colon cancer” in their NCBI record</li>
<li>“Homo sapiens”[ORGN] AND malaria - Which are from human, and contain “malaria” in their NCBI record</li>
<li>“Homo sapiens”[ORGN] AND biomol_mrna[PROP] - Which are mRNA sequences from human</li>
<li>“Bacteria”[ORGN] AND srcdb_refseq[PROP] - Which are RefSeq sequences from Bacteria</li>
<li>“colon cancer” AND srcdb_refseq[PROP] - From RefSeq, which contain “colon cancer” in their NCBI record</li>
</ul>
<p>Note that if you are searching for a phrase such as “colon cancer” or “Chlamydia trachomatis”, you need to put the phrase in quotes when typing it into the search box. This is because if you type the phrase in the search box without quotes, the search will be for NCBI records that contain either of the two words “colon” or “cancer” (or either of the two words “Chlamydia” or “trachomatis”), not necessarily both words.</p>
<p>As mentioned above, the NCBI database contains several sub-databases, including the NCBI <strong>Nucleotide database</strong> and the NCBI <strong>Protein database</strong>. If you go to the NCBI website, and type one of the search queries above in the search box at the top of the page, the results page will tell you how many matching NCBI records were found in each of the NCBI sub-databases.</p>
<p>For example, if you search for “Chlamydia trachomatis[ORGN]”, you will get matches to proteins from C. trachomatis in the NCBI Protein database, matches to DNA and RNA sequences from <em>C. trachomatis</em> in the NCBI Nucleotide database, matches to whole genome sequences for C. trachomatis strains in the NCBI Genome database, and so on:</p>
<p>Alternatively, if you know in advance that you want to search a particular sub-database, for example, the NCBI Protein database, when you go to the NCBI website, you can select that sub-database from the drop-down list above the search box, so that you will search that sub-database.</p>
</div>
<div id="example-finding-the-sequences-published-in-nature-460352-358-for-reference" class="section level2">
<h2><span class="header-section-number">8.9</span> Example: finding the sequences published in Nature 460:352-358 (for reference)</h2>
<p><strong>NOTE</strong>: The following section is here for reference; you need to know its possible to refine searches but do not need to know any of these actual tags.</p>
<p>For example, if you want to find sequences published in Nature 460:352-358, you can use the “[JOUR]”, “[VOL]” and “[PAGE]” search terms. That is, you would go to the NCBI website and type in the search box on the top: “Nature”[JOUR] AND 460[VOL] AND 352[PAGE], where [JOUR] specifies the journal name, [VOL] the volume of the journal the paper is in, and [PAGE] the page number.</p>
<p>This should bring up a results page with “50890” beside the word “Nucleotide”, and “1” beside the word “Genome”, and “25701” beside the word “Protein”, indicating that there were 50890 hits to sequence records in the Nucleotide database, which contains DNA and RNA sequences, and 1 hit to the Genome database, which contains genome sequences, and 25701 hits to the Protein database, which contains protein sequences.</p>
<p>If you click on the word “Nucleotide”, it will bring up a webpage with a list of links to the NCBI sequence records for those 50890 hits. The 50890 hits are all contigs from the schistosome worm <em>Schistosoma mansoni</em>.</p>
<p>Likewise, if you click on the word “Protein”, it will bring up a webpage with a list of links to the NCBI sequence records for the 25701 hits, and you will see that the hits are all predicted proteins for <em>Schistosoma mansoni</em>.</p>
<p>If you click on the word “Genome”, it will bring you to the NCBI record for the <em>Schistosoma mansoni</em> genome sequence, which has NCBI accession NS_00200. Note that the accession starts with “NS_”, which indicates that it is a RefSeq accession.</p>
<p>Therefore, in Nature volume 460, page 352, the <em>Schistosoma mansoni</em> genome sequence was published, along with all the DNA sequence contigs that were sequenced for the genome project, and all the predicted proteins for the gene predictions made in the genome sequence. You can view the original paper on the Nature website at <a href="http://www.nature.com/nature/journal/v460/n7253/abs/nature08160.html" class="uri">http://www.nature.com/nature/journal/v460/n7253/abs/nature08160.html</a>.</p>
<p>Note: <em>Schistmosoma mansoni</em> is a parasitic worm that is responsible for causing <strong>schistosomiasis</strong>, which is classified by the WHO as a <strong>neglected tropical disease</strong>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="downloading-ncbi-sequence-data-by-hand.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="downloading-sequences-from-uniprot-by-hand.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["compbio4all.pdf", "compbio4all.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
