<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Local variation in GC content - NOT UPDATED | Computational Biology for All</title>
  <meta name="description" content="A bioinformatics and computational biology book" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Local variation in GC content - NOT UPDATED | Computational Biology for All" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A bioinformatics and computational biology book" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Local variation in GC content - NOT UPDATED | Computational Biology for All" />
  
  <meta name="twitter:description" content="A bioinformatics and computational biology book" />
  

<meta name="author" content="Avril Coglan and Nathan Brouwer" />


<meta name="date" content="2021-01-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sequence-databases.html"/>
<link rel="next" href="alignments-in-r-not-updayed.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Computational Biology for All</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#changes-made-to-original-source"><i class="fa fa-check"></i><b>1.1</b> Changes made to original source</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="downloading-r.html"><a href="downloading-r.html"><i class="fa fa-check"></i><b>2</b> Downloading R</a><ul>
<li class="chapter" data-level="2.1" data-path="downloading-r.html"><a href="downloading-r.html#introduction-to-r"><i class="fa fa-check"></i><b>2.1</b> Introduction to R</a></li>
<li class="chapter" data-level="2.2" data-path="downloading-r.html"><a href="downloading-r.html#installing-r"><i class="fa fa-check"></i><b>2.2</b> Installing R</a><ul>
<li class="chapter" data-level="2.2.1" data-path="downloading-r.html"><a href="downloading-r.html#installing-r-on-a-windows-pc"><i class="fa fa-check"></i><b>2.2.1</b> Installing R on a Windows PC</a></li>
<li class="chapter" data-level="2.2.2" data-path="downloading-r.html"><a href="downloading-r.html#how-to-install-r-on-non-windows-computers-eg.-macintosh-or-linux-computers"><i class="fa fa-check"></i><b>2.2.2</b> How to install R on non-Windows computers (eg. Macintosh or Linux computers)</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="downloading-r.html"><a href="downloading-r.html#starting-r"><i class="fa fa-check"></i><b>2.3</b> Starting R</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="downloading-rstudio.html"><a href="downloading-rstudio.html"><i class="fa fa-check"></i><b>3</b> Downloading RStudio</a><ul>
<li class="chapter" data-level="3.1" data-path="downloading-rstudio.html"><a href="downloading-rstudio.html#installing-rstudio"><i class="fa fa-check"></i><b>3.1</b> Installing RStudio</a></li>
<li class="chapter" data-level="3.2" data-path="downloading-rstudio.html"><a href="downloading-rstudio.html#getting-to-know-rstudio"><i class="fa fa-check"></i><b>3.2</b> Getting to know RStudio</a></li>
<li class="chapter" data-level="3.3" data-path="downloading-rstudio.html"><a href="downloading-rstudio.html#rstudio-versus-rstudio-cloud"><i class="fa fa-check"></i><b>3.3</b> RStudio versus RStudio Cloud</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="installing-r-packages.html"><a href="installing-r-packages.html"><i class="fa fa-check"></i><b>4</b> Installing R packages</a><ul>
<li class="chapter" data-level="4.1" data-path="installing-r-packages.html"><a href="installing-r-packages.html#downloading-packages-with-the-rstudio-gui"><i class="fa fa-check"></i><b>4.1</b> Downloading packages with the RStudio GUI</a></li>
<li class="chapter" data-level="4.2" data-path="installing-r-packages.html"><a href="installing-r-packages.html#downloading-packages-with-the-function-install.packages"><i class="fa fa-check"></i><b>4.2</b> Downloading packages with the function install.packages()</a></li>
<li class="chapter" data-level="4.3" data-path="installing-r-packages.html"><a href="installing-r-packages.html#using-packages-after-they-are-downloaded"><i class="fa fa-check"></i><b>4.3</b> Using packages after they are downloaded</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html"><i class="fa fa-check"></i><b>5</b> Installing Bioconductor</a><ul>
<li class="chapter" data-level="5.1" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html#bioconductor"><i class="fa fa-check"></i><b>5.1</b> Bioconductor</a></li>
<li class="chapter" data-level="5.2" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html#installing-biocmanager"><i class="fa fa-check"></i><b>5.2</b> Installing BiocManager</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html"><i class="fa fa-check"></i><b>6</b> Brief introduction to R</a><ul>
<li class="chapter" data-level="6.1" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#vocabulary"><i class="fa fa-check"></i><b>6.1</b> Vocabulary</a></li>
<li class="chapter" data-level="6.2" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#r-functions-etc"><i class="fa fa-check"></i><b>6.2</b> R functions etc</a></li>
<li class="chapter" data-level="6.3" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#a-brief-introduction-to-r"><i class="fa fa-check"></i><b>6.3</b> A brief introduction to R</a></li>
<li class="chapter" data-level="6.4" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#links-and-further-reading"><i class="fa fa-check"></i><b>6.4</b> Links and Further Reading</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html"><i class="fa fa-check"></i><b>7</b> Introduction to NCBI databases</a><ul>
<li class="chapter" data-level="7.1" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#to-do"><i class="fa fa-check"></i><b>7.1</b> To do</a></li>
<li class="chapter" data-level="7.2" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#the-ncbi-sequence-database"><i class="fa fa-check"></i><b>7.2</b> The NCBI Sequence Database</a></li>
<li class="chapter" data-level="7.3" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#the-ncbi-sub-databases"><i class="fa fa-check"></i><b>7.3</b> The NCBI Sub-Databases</a></li>
<li class="chapter" data-level="7.4" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#searching-for-an-accession-number-in-the-ncbi-database"><i class="fa fa-check"></i><b>7.4</b> Searching for an accession number in the NCBI database</a></li>
<li class="chapter" data-level="7.5" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#ncbi-sequence-format-ncbi-format"><i class="fa fa-check"></i><b>7.5</b> NCBI Sequence Format (NCBI Format)</a></li>
<li class="chapter" data-level="7.6" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#refseq"><i class="fa fa-check"></i><b>7.6</b> RefSeq</a></li>
<li class="chapter" data-level="7.7" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#querying-the-ncbi-database"><i class="fa fa-check"></i><b>7.7</b> Querying the NCBI Database</a></li>
<li class="chapter" data-level="7.8" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#querying-the-ncbi-database-via-the-ncbi-website-for-reference"><i class="fa fa-check"></i><b>7.8</b> Querying the NCBI Database via the NCBI Website (for reference)</a></li>
<li class="chapter" data-level="7.9" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#example-finding-the-sequences-published-in-nature-460352-358-for-reference"><i class="fa fa-check"></i><b>7.9</b> Example: finding the sequences published in Nature 460:352-358 (for reference)</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="downloading-ncbi-sequence-data-by-hand.html"><a href="downloading-ncbi-sequence-data-by-hand.html"><i class="fa fa-check"></i><b>8</b> Downloading NCBI sequence data by hand</a><ul>
<li class="chapter" data-level="8.1" data-path="downloading-ncbi-sequence-data-by-hand.html"><a href="downloading-ncbi-sequence-data-by-hand.html#retrieving-genome-sequence-data-via-the-ncbi-website"><i class="fa fa-check"></i><b>8.1</b> Retrieving genome sequence data via the NCBI website</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html"><i class="fa fa-check"></i><b>9</b> Downloading sequences from UniProt by hand</a><ul>
<li class="chapter" data-level="9.1" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#vocab"><i class="fa fa-check"></i><b>9.1</b> Vocab</a></li>
<li class="chapter" data-level="9.2" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#uniprot"><i class="fa fa-check"></i><b>9.2</b> UniProt</a></li>
<li class="chapter" data-level="9.3" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#viewing-the-uniprot-webpage-for-a-protein-sequence"><i class="fa fa-check"></i><b>9.3</b> Viewing the UniProt webpage for a protein sequence</a><ul>
<li class="chapter" data-level="9.3.1" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#protein-function"><i class="fa fa-check"></i><b>9.3.1</b> Protein function</a></li>
<li class="chapter" data-level="9.3.2" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#protein-sequence-and-size"><i class="fa fa-check"></i><b>9.3.2</b> Protein sequence and size</a></li>
<li class="chapter" data-level="9.3.3" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#other-information"><i class="fa fa-check"></i><b>9.3.3</b> Other information</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#retrieving-a-uniprot-protein-sequence-via-the-uniprot-website"><i class="fa fa-check"></i><b>9.4</b> Retrieving a UniProt protein sequence via the UniProt website</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="sequence-databases.html"><a href="sequence-databases.html"><i class="fa fa-check"></i><b>10</b> Sequence Databases</a><ul>
<li class="chapter" data-level="10.1" data-path="sequence-databases.html"><a href="sequence-databases.html#querying-the-ncbi-database-via-r"><i class="fa fa-check"></i><b>10.1</b> Querying the NCBI Database via R</a></li>
<li class="chapter" data-level="10.2" data-path="sequence-databases.html"><a href="sequence-databases.html#summary"><i class="fa fa-check"></i><b>10.2</b> Summary</a></li>
<li class="chapter" data-level="10.3" data-path="sequence-databases.html"><a href="sequence-databases.html#acknowledgements"><i class="fa fa-check"></i><b>10.3</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html"><i class="fa fa-check"></i><b>11</b> Local variation in GC content - NOT UPDATED</a><ul>
<li class="chapter" data-level="11.0.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#note-on-the-biology-in-this-section"><i class="fa fa-check"></i><b>11.0.1</b> Note on the biology in this section</a></li>
<li class="chapter" data-level="11.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#vocabulary-1"><i class="fa fa-check"></i><b>11.1</b> Vocabulary</a></li>
<li class="chapter" data-level="11.2" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#reading-sequence-data-with-rentrezentrez_fetch"><i class="fa fa-check"></i><b>11.2</b> Reading sequence data with rentrez::entrez_fetch</a></li>
<li class="chapter" data-level="11.3" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#local-variation-in-gc-content"><i class="fa fa-check"></i><b>11.3</b> Local variation in GC content</a></li>
<li class="chapter" data-level="11.4" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#acknowledgements-1"><i class="fa fa-check"></i><b>11.4</b> Acknowledgements</a><ul>
<li class="chapter" data-level="11.4.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#license"><i class="fa fa-check"></i><b>11.4.1</b> License</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#exercises"><i class="fa fa-check"></i><b>11.5</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html"><i class="fa fa-check"></i><b>12</b> Alignments in R - NOT UPDAYED</a><ul>
<li class="chapter" data-level="12.1" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#retrieving-a-uniprot-protein-sequence-using-seqinr"><i class="fa fa-check"></i><b>12.1</b> Retrieving a UniProt protein sequence using SeqinR</a></li>
<li class="chapter" data-level="12.2" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#comparing-two-sequences-using-a-dotplot"><i class="fa fa-check"></i><b>12.2</b> Comparing two sequences using a dotplot</a></li>
<li class="chapter" data-level="12.3" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#aligning-uniprot-sequences"><i class="fa fa-check"></i><b>12.3</b> Aligning UniProt sequences</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html"><i class="fa fa-check"></i><b>13</b> Multiple alignment with Clustal</a><ul>
<li class="chapter" data-level="13.1" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#installing-the-clustal-multiple-alignment-software"><i class="fa fa-check"></i><b>13.1</b> Installing the CLUSTAL multiple alignment software</a></li>
<li class="chapter" data-level="13.2" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#discarding-very-poorly-conserved-regions-from-an-alignment"><i class="fa fa-check"></i><b>13.2</b> Discarding very poorly conserved regions from an alignment</a></li>
<li class="chapter" data-level="13.3" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#calculating-genetic-distances-between-protein-sequences"><i class="fa fa-check"></i><b>13.3</b> Calculating genetic distances between protein sequences</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html"><i class="fa fa-check"></i><b>14</b> Unrooted Neighbor-joining Phylogenetic trees</a><ul>
<li class="chapter" data-level="14.1" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#building-an-unrooted-phylogenetic-tree-for-protein-sequences"><i class="fa fa-check"></i><b>14.1</b> Building an unrooted phylogenetic tree for protein sequences</a></li>
<li class="chapter" data-level="14.2" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#calculating-the-statistical-significance-of-a-pairwise-global-alignment"><i class="fa fa-check"></i><b>14.2</b> Calculating the statistical significance of a pairwise global alignment</a></li>
<li class="chapter" data-level="14.3" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#summary-1"><i class="fa fa-check"></i><b>14.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html"><i class="fa fa-check"></i><b>15</b> Retrieving multiple sequences in R</a><ul>
<li class="chapter" data-level="15.1" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html#prelminaries"><i class="fa fa-check"></i><b>15.1</b> Prelminaries</a></li>
<li class="chapter" data-level="15.2" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html#retrieving-a-list-of-sequences-from-uniprot"><i class="fa fa-check"></i><b>15.2</b> Retrieving a list of sequences from UniProt</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html"><i class="fa fa-check"></i><b>16</b> Computational gene finding</a><ul>
<li class="chapter" data-level="16.1" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#the-genetic-code"><i class="fa fa-check"></i><b>16.1</b> The genetic code</a></li>
<li class="chapter" data-level="16.2" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#reading-frames"><i class="fa fa-check"></i><b>16.2</b> Reading frames</a></li>
<li class="chapter" data-level="16.3" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#finding-open-reading-frames-on-the-forward-strand-of-a-dna-sequence"><i class="fa fa-check"></i><b>16.3</b> Finding open reading frames on the forward strand of a DNA sequence</a></li>
<li class="chapter" data-level="16.4" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#lengths-of-open-reading-frames"><i class="fa fa-check"></i><b>16.4</b> Lengths of open reading frames</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html"><i class="fa fa-check"></i><b>17</b> Significance of ORFs</a><ul>
<li class="chapter" data-level="17.1" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html#identifying-significant-open-reading-frames"><i class="fa fa-check"></i><b>17.1</b> Identifying significant open reading frames</a></li>
<li class="chapter" data-level="17.2" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html#summary-2"><i class="fa fa-check"></i><b>17.2</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="rooted-neighbor-joing-phylogenetic-trees.html"><a href="rooted-neighbor-joing-phylogenetic-trees.html"><i class="fa fa-check"></i><b>18</b> Rooted Neighbor-joing Phylogenetic trees</a><ul>
<li class="chapter" data-level="18.1" data-path="rooted-neighbor-joing-phylogenetic-trees.html"><a href="rooted-neighbor-joing-phylogenetic-trees.html#building-a-rooted-phylogenetic-tree-for-protein-sequences"><i class="fa fa-check"></i><b>18.1</b> Building a rooted phylogenetic tree for protein sequences</a></li>
<li class="chapter" data-level="18.2" data-path="rooted-neighbor-joing-phylogenetic-trees.html"><a href="rooted-neighbor-joing-phylogenetic-trees.html#building-a-phylogenetic-tree-for-dna-or-mrna-sequences"><i class="fa fa-check"></i><b>18.2</b> Building a phylogenetic tree for DNA or mRNA sequences</a></li>
<li class="chapter" data-level="18.3" data-path="rooted-neighbor-joing-phylogenetic-trees.html"><a href="rooted-neighbor-joing-phylogenetic-trees.html#summary-3"><i class="fa fa-check"></i><b>18.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="todo.html"><a href="todo.html"><i class="fa fa-check"></i><b>19</b> TODO</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://https://github.com/brouwern/CB4ALL" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Computational Biology for All</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="local-variation-in-gc-content---not-updated" class="section level1">
<h1><span class="header-section-number">Chapter 11</span> Local variation in GC content - NOT UPDATED</h1>
<p>By: Avril Coghlan, adapted by Nathan Brouwer</p>
<p><strong>TODO</strong></p>
<p>need fasta clearn function put into biodata</p>
<p>Preliminaries</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="local-variation-in-gc-content-not-updated.html#cb53-1"></a><span class="kw">library</span>(rentrez)</span>
<span id="cb53-2"><a href="local-variation-in-gc-content-not-updated.html#cb53-2"></a><span class="kw">library</span>(seqinr)</span></code></pre></div>
<div id="note-on-the-biology-in-this-section" class="section level3">
<h3><span class="header-section-number">11.0.1</span> Note on the biology in this section</h3>
<p>Some of the biology in this tutorial appears to be out of date. For example, using variation in GC content to ID horizontal gene transfer is currently considered to be biased. The examples are still good for practicing R skills.</p>
</div>
<div id="vocabulary-1" class="section level2">
<h2><span class="header-section-number">11.1</span> Vocabulary</h2>
<p><a href="https://en.wikipedia.org/wiki/GC-content"><strong>GC content</strong></a>
<a href="en.wikipedia.org/wiki/Horizontal_gene_transfer"><strong>horizontal transfer</strong></a>
<strong>local variation</strong> in GC content</p>
</div>
<div id="reading-sequence-data-with-rentrezentrez_fetch" class="section level2">
<h2><span class="header-section-number">11.2</span> Reading sequence data with rentrez::entrez_fetch</h2>
<p>In a previous section you learned how to use to search for and download the sequence data for a given NCBI accession from the NCBI Sequence Database, either via the NCBI website using entrez_fetch() from the rentrez package.</p>
<p>For example, you could have downloaded the sequence data for a the DEN-1 Dengue virus sequence (NCBI accession NC_001477), and stored it on a file on your computer (eg. “dengue_fasta.fasta”).</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="local-variation-in-gc-content-not-updated.html#cb54-1"></a>dengueseq_fasta &lt;-<span class="st"> </span><span class="kw">entrez_fetch</span>(<span class="dt">db =</span> <span class="st">&quot;nucleotide&quot;</span>, </span>
<span id="cb54-2"><a href="local-variation-in-gc-content-not-updated.html#cb54-2"></a>                          <span class="dt">id =</span> <span class="st">&quot;NC_001477&quot;</span>, </span>
<span id="cb54-3"><a href="local-variation-in-gc-content-not-updated.html#cb54-3"></a>                          <span class="dt">rettype =</span> <span class="st">&quot;fasta&quot;</span>)</span></code></pre></div>
<p>As noted before, the file gets downloaded in FASTA format, which isn’t directly useable in R</p>
<p>We can convert our FASTA object into a vector using the function fasta_cleaner()</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="local-variation-in-gc-content-not-updated.html#cb55-1"></a>header. &lt;-<span class="st"> &quot;&gt;NC_001477.1 Dengue virus 1, complete genome&quot;</span></span>
<span id="cb55-2"><a href="local-variation-in-gc-content-not-updated.html#cb55-2"></a>dengueseq_vector &lt;-<span class="st"> </span><span class="kw">fasta_cleaner</span>(dengueseq_fasta)</span></code></pre></div>
<p>Once you have retrieved a sequence from the NCBI Sequence Database and stored it in a vector variable such as dengueseq_vector in the example above, it is possible to extract <strong>subsequence</strong> of the sequence by typing the name of the vector (eg. dengueseq_vector) followed by the square brackets containing the indices for those nucleotides. For example, to obtain nucleotides 452-535 of the DEN-1 Dengue virus genome, we can type:</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="local-variation-in-gc-content-not-updated.html#cb56-1"></a>dengueseq_vector[<span class="dv">452</span><span class="op">:</span><span class="dv">535</span>]</span></code></pre></div>
</div>
<div id="local-variation-in-gc-content" class="section level2">
<h2><span class="header-section-number">11.3</span> Local variation in GC content</h2>
<p>In a previous section, you learned that to find out the <a href="https://en.wikipedia.org/wiki/GC-content"><strong>GC content</strong></a> of a genome sequence (percentage of nucleotides in a genome sequence that are Gs or Cs), you can use the GC() function in the SeqinR package. For example, to find the GC content of the DEN-1 Dengue virus sequence that we have stored in the vector dengueseq, we can type:</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="local-variation-in-gc-content-not-updated.html#cb57-1"></a><span class="kw">GC</span>(dengueseq_vector)</span></code></pre></div>
<p>The output of the GC() is the fraction of nucleotides in a sequence that are Gs or Cs, so to convert it to a percentage we need to multiply by 100.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="local-variation-in-gc-content-not-updated.html#cb58-1"></a><span class="kw">GC</span>(dengueseq_vector)<span class="op">*</span><span class="dv">100</span></span></code></pre></div>
<p>Thus, the GC content of the DEN-1 Dengue virus genome is about 0.467 or 46.7%.</p>
<p>Although the GC content of the whole DEN-1 Dengue virus genome sequence is about 46.7%, there is probably <strong>local variation</strong> in GC content within the genome. That is, some regions of the genome sequence may have GC contents quite a bit higher than 46.7%, while some regions of the genome sequence may have GC contents that are quite a big lower than 46.7%. Local fluctuations in GC content within the genome sequence can provide different interesting information, for example, they may reveal cases of <a href="en.wikipedia.org/wiki/Horizontal_gene_transfer"><strong>horizontal transfer</strong></a> or reveal biases in mutation.</p>
<p>If a chunk of DNA has moved by horizontal transfer from the genome of a species with low GC content to a species with high GC content, the chunk of horizontally transferred DNA could be detected as a region of unusually low GC content in the high-GC recipient genome.</p>
<p>On the other hand, a region unusually low GC content in an otherwise high-GC content genome could also arise due to biases in mutation in that region of the genome, for example, if mutations from Gs/Cs to Ts/As are more common for some reason in that region of the genome than in the rest of the genome.</p>
<p>A sliding window analysis of GC content
In order to study local variation in GC content within a genome sequence, we could calculate the GC content for small chunks of the genome sequence. The DEN-1 Dengue virus genome sequence is 10735 nucleotides long. To study variation in GC content within the genome sequence, we could calculate the GC content of chunks of the DEN-1 Dengue virus genome, for example, for each 2000-nucleotide chunk of the genome sequence:</p>
<p>GC(dengueseq[1:2000]) # Calculate the GC content of nucleotides 1-2000 of the Dengue genome</p>
<p>GC(dengueseq[2001:4000]) # Calculate the GC content of nucleotides 2001-4000 of the Dengue genome</p>
<p>From the output of the above calculations, we see that the region of the DEN-1 Dengue virus genome from nucleotides 1-2000 has a GC content of 46.5%, while the region of the Dengue genome from nucleotides 2001-4000 has a GC content of about 45.3%. Thus, there seems to be some local variation in GC content within the Dengue genome sequence.</p>
<p>Instead of typing in the commands above to tell R to calculate the GC content for each 2000-nucleotide chunk of the DEN-1 Dengue genome, we can use a for loop to carry out the same calculations, but by typing far fewer commands. That is, we can use a for loop to take each 2000-nucleotide chunk of the DEN-1 Dengue virus genome, and to calculate the GC content of each 2000-nucleotide chunk. Below we will explain the following for loop that has been written for this purpose:</p>
<p>starts &lt;- seq(1, length(dengueseq)-2000, by = 2000)
starts</p>
<p>n &lt;- length(starts) # Find the length of the vector “starts”
for (i in 1:n) {
chunk &lt;- dengueseq[starts[i]:(starts[i]+1999)]
chunkGC &lt;- GC(chunk)
print (chunkGC)
}</p>
<p>The command “starts &lt;- seq(1, length(dengueseq)-2000, by = 2000)” stores the result of the seq() command in the vector starts, which contains the values 1, 2001, 4001, 6001, and 8001.</p>
<p>We set the variable n to be equal to the number of elements in the vector starts, so it will be 5 here, since the vector starts contains the five elements 1, 2001, 4001, 6001 and 8001. The line “for (i in 1:n)” means that the counter i will take values of 1-5 in subsequent cycles of the for loop. The for loop above is spread over several lines. However, R will not execute the commands within the for loop until you have typed the final “}” at the end of the for loop and pressed “Return”.</p>
<p>Each of the three commands within the for loop are carried out in each cycle of the loop. In the first cycle of the loop, i is 1, the vector variable chunk is used to store the region from nucleotides 1-2000 of the Dengue virus sequence, the GC content of that region is calculated and stored in the variable chunkGC, and the value of chunkGC is printed out.</p>
<p>In the second cycle of the loop, i is 2, the vector variable chunk is used to store the region from nucleotides 2001-4000 of the Dengue virus sequence, the GC content of that region is calculated and stored in the variable chunkGC, and the value of chunkGC is printed out. The loop continues until the value of i is 5. In the fifth cycle through the loop, the value of i is 5, and so the GC content of the region from nucleotides 8001-10000 is printed out.</p>
<p>Note that we stop the loop when we are looking at the region from nucleotides 8001-10000, instead of continuing to another cycle of the loop where the region under examiniation would be from nucleotides 10001-12000. The reason for this is because the length of the Dengue virus genome sequence is just 10735 nucleotides, so there is not a full 2000-nucleotide region from nucleotide 10001 to the end of the sequence at nucleotide 10735.</p>
<p>The above analysis of local variation in GC content is what is known as a sliding window analysis of GC content. By calculating the GC content in each 2000-nucleotide chunk of the Dengue virus genome, you are effectively sliding a 2000-nucleotide window along the DNA sequence from start to end, and calculating the GC content in each non-overlapping window (chunk of DNA).</p>
<p>Note that this sliding window analysis of GC content is a slightly simplified version of the method usually carried out by bioinformaticians. In this simplified version, we have calculated the GC content in non-overlapping windows along a DNA sequence. However, it is more usual to calculate GC content in overlapping windows along a sequence, although that makes the code slightly more complicated.</p>
<p>A sliding window plot of GC content
It is common to use the data generated from a sliding window analysis to create a sliding window plot of GC content. To create a sliding window plot of GC content, you plot the local GC content in each window of the genome, versus the nucleotide position of the start of each window. We can create a sliding window plot of GC content by typing:</p>
<p>starts &lt;- seq(1, length(dengueseq)-2000, by = 2000)
n &lt;- length(starts) # Find the length of the vector “starts”
chunkGCs &lt;- numeric(n) # Make a vector of the same length as vector “starts”, but just containing zeroes
for (i in 1:n) {
chunk &lt;- dengueseq[starts[i]:(starts[i]+1999)]
chunkGC &lt;- GC(chunk)
print(chunkGC)
chunkGCs[i] &lt;- chunkGC
}
plot(starts,chunkGCs,type=“b”,xlab=“Nucleotide start position”,ylab=“GC content”)
image2</p>
<p>In the code above, the line “chunkGCs &lt;- numeric(n)” makes a new vector chunkGCs which has the same number of elements as the vector starts (5 elements here). This vector chunkGCs is then used within the for loop for storing the GC content of each chunk of DNA.</p>
<p>After the loop, the vector starts can be plotted against the vector chunkGCs using the plot() function, to get a plot of GC content against nucleotide position in the genome sequence. This is a sliding window plot of GC content.</p>
<p>You may want to use the code above to create sliding window plots of GC content of different species’ genomes, using different windowsizes. Therefore, it makes sense to write a function to do the sliding window plot, that can take the windowsize that the user wants to use and the sequence that the user wants to study as arguments (inputs):</p>
<p>slidingwindowplot &lt;- function(windowsize, inputseq)
{
starts &lt;- seq(1, length(inputseq)-windowsize, by = windowsize)
n &lt;- length(starts) # Find the length of the vector “starts”
chunkGCs &lt;- numeric(n) # Make a vector of the same length as vector “starts”, but just containing zeroes
for (i in 1:n) {
chunk &lt;- inputseq[starts[i]:(starts[i]+windowsize-1)]
chunkGC &lt;- GC(chunk)
print(chunkGC)
chunkGCs[i] &lt;- chunkGC
}
plot(starts,chunkGCs,type=“b”,xlab=“Nucleotide start position”,ylab=“GC content”)
}
This function will make a sliding window plot of GC content for a particular input sequence inputseq specified by the user, using a particular windowsize windowsize specified by the user. Once you have typed in this function once, you can use it again and again to make sliding window plots of GC contents for different input DNA sequences, with different windowsizes. For example, you could create two different sliding window plots of the DEN-1 Dengue virus genome sequence, using windowsizes of 3000 and 300 nucleotides, respectively:</p>
<p>slidingwindowplot(3000, dengueseq)
image3</p>
<p>slidingwindowplot(300, dengueseq)
image4</p>
<p>Over-represented and under-represented DNA words
In the previous chapter, you learned that the count() function in the SeqinR R package can calculate the frequency of all DNA words of a certain length in a DNA sequence. For example, if you want to know the frequency of all DNA words that are 2 nucleotides long in the Dengue virus genome sequence, you can type:</p>
<p>count(dengueseq, 2)
aa ac ag at ca cc cg ct ga gc gg gt ta tc tg tt
1108 720 890 708 901 523 261 555 976 500 787 507 440 497 832 529
It is interesting to identify DNA words that are two nucleotides long (“dinucleotides”, ie. “AT”, “AC”, etc.) that are over-represented or under-represented in a DNA sequence. If a particular DNA word is over-represented in a sequence, it means that it occurs many more times in the sequence than you would have expected by chance. Similarly, if a particular DNA word is under-represented in a sequence, it means it occurs far fewer times in the sequence than you would have expected.</p>
<p>A statistic called ρ (Rho) is used to measure how over- or under-represented a particular DNA word is. For a 2-nucleotide (dinucleotide) DNA word ρ is calculated as:</p>
<p>ρ(xy) = fxy/(fx*fy),</p>
<p>where fxy and fx are the frequencies of the DNA words xy and x in the DNA sequence under study. For example, the value of ρ for the DNA word “TA” can be calculated as: ρ(TA) = fTA/(fT* fA), where fTA, fT and fA are the frequencies of the DNA words “TA”, “T” and “A” in the DNA sequence.</p>
<p>The idea behind the ρ statistic is that, if a DNA sequence had a frequency fx of a 1-nucleotide DNA word x, and a frequency fy of a 1-nucleotide DNA word y, then we expect the frequency of the 2-nucleotide DNA word xy to be fx* fy. That is, the frequencies of the 2-nucleotide DNA words in a sequence are expected to be equal the products of the specific frequencies of the two nucleotides that compose them. If this were true, then ρ would be equal to 1. If we find that ρ is much greater than 1 for a particular 2-nucleotide word in a sequence, it indicates that that 2-nucleotide word is much more common in that sequence than expected (ie. it is over-represented).</p>
<p>For example, say that your input sequence has only 5% Ts (ie. fT = 0.05). In a random DNA sequence with 5% Ts, you would expect to see the word “TT” very infrequently. In fact, we would only expect 0.05 * 0.05=0.0025 (0.25%) of 2-nucleotide words to be TTs (ie. we expect fTT = fT* fT). This is because Ts are rare, so they are expected to be adjacent to each other very infrequently if the few Ts are randomly scattered throughout the DNA. Therefore, if you see lots of TT 2-nucleotide words in your real input sequence (eg. fTT = 0.3, so ρ = 0.3/0.0025 = 120), you would suspect that natural selection has acted to increase the number of occurrences of the TT word in the sequence (presumably because it has some beneficial biological function).</p>
<p>To find over-represented and under-represented DNA words that are 2 nucleotides long in the DEN-1 Dengue virus sequence, we can calculate the ρ statistic for each 2-nucleotide word in the sequence. For example, given the number of occurrences of the individual nucleotides A, C, G and T in the Dengue sequence, and the number of occurrences of the DNA word GC in the sequence (500, from above), we can calculate the value of ρ for the 2-nucleotide DNA word “GC”, using the formula ρ(GC) = fGC/(fG * fC), where fGC, fG and fC are the frequencies of the DNA words “GC”, “G” and “C” in the DNA sequence:</p>
<p>count(dengueseq, 1) # Get the number of occurrences of 1-nucleotide DNA words
a c g t
3426 2240 2770 2299
2770/(3426+2240+2770+2299) # Get fG
[1] 0.2580345
2240/(3426+2240+2770+2299) # Get fC
[1] 0.2086633
count(dengueseq, 2) # Get the number of occurrences of 2-nucleotide DNA words
aa ac ag at ca cc cg ct ga gc gg gt ta tc tg tt
1108 720 890 708 901 523 261 555 976 500 787 507 440 497 832 529
500/(1108+720+890+708+901+523+261+555+976+500+787+507+440+497+832+529) # Get fGC
[1] 0.04658096
0.04658096/(0.2580345*0.2086633) # Get rho(GC)
[1] 0.8651364
We calculate a value of ρ(GC) of approximately 0.865. This means that the DNA word “GC” is about 0.865 times as common in the DEN-1 Dengue virus sequence than expected. That is, it seems to be slightly under-represented.</p>
<p>Note that if the ratio of the observed to expected frequency of a particular DNA word is very low or very high, then we would suspect that there is a statistical under-representation or over-representation of that DNA word. However, to be sure that this over- or under-representation is statistically significant, we would need to do a statistical test. We will not deal with the topic of how to carry out the statistical test here.</p>
<p>Summary
In this chapter, you will have learned to use the following R functions:</p>
<p>seq() for creating a sequence of numbers
print() for printing out the value of a variable
plot() for making a plot (eg. a scatterplot)
numeric() for making a numeric vector of a particular length
function() for making a function
All of these functions belong to the standard installation of R. You also learned how to use for loops to carry out the same operation again and again, each time on different inputs.</p>
<p>Links and Further Reading
Some links are included here for further reading.</p>
<p>For background reading on DNA sequence statistics, it is recommended to read Chapter 1 of Introduction to Computational Genomics: a case studies approach by Cristianini and Hahn (Cambridge University Press; www.computational-genomics.net/book/).</p>
<p>For more in-depth information and more examples on using the SeqinR package for sequence analysis, look at the SeqinR documentation, <a href="http://pbil.univ-lyon1.fr/software/seqinr/doc.php?lang=eng" class="uri">http://pbil.univ-lyon1.fr/software/seqinr/doc.php?lang=eng</a>.</p>
<p>There is also a very nice chapter on “Analyzing Sequences”, which includes examples of using SeqinR for sequence analysis, in the book Applied statistics for bioinformatics using R by Krijnen (available online at cran.r-project.org/doc/contrib/Krijnen-IntroBioInfStatistics.pdf).</p>
<p>For a more in-depth introduction to R, a good online tutorial is available on the “Kickstarting R” website, cran.r-project.org/doc/contrib/Lemon-kickstart.</p>
<p>There is another nice (slightly more in-depth) tutorial to R available on the “Introduction to R” website, cran.r-project.org/doc/manuals/R-intro.html.</p>
<p>Acknowledgements
Thank you to Noel O’Boyle for helping in using Sphinx, <a href="http://sphinx.pocoo.org" class="uri">http://sphinx.pocoo.org</a>, to create this document, and github, <a href="https://github.com/" class="uri">https://github.com/</a>, to store different versions of the document as I was writing it, and readthedocs, <a href="http://readthedocs.org/" class="uri">http://readthedocs.org/</a>, to build and distribute this document.</p>
</div>
<div id="acknowledgements-1" class="section level2">
<h2><span class="header-section-number">11.4</span> Acknowledgements</h2>
<p>This is a modification of <a href="https://a-little-book-of-r-for-bioinformatics.readthedocs.io/en/latest/src/chapter1.html">“DNA Sequence Statistics (1)”</a> from Avril Coghlan’s <a href="https://a-little-book-of-r-for-bioinformatics.readthedocs.io/en/latest/index.html"><em>A little book of R for bioinformatics.</em></a>. Almost all of text and code was originally written by Dr. Coghlan and distributed under the <a href="https://creativecommons.org/licenses/by/3.0/us/">Creative Commons 3.0</a> license.</p>
<p>In “A little book…” Coghlan write “Many of the ideas for the examples and exercises for this chapter were inspired by the Matlab case studies on Haemophilus influenzae (www.computational-genomics.net/case_studies/haemophilus_demo.html) and Bacteriophage lambda (<a href="http://www.computational-genomics.net/case_studies/lambdaphage_demo.html" class="uri">http://www.computational-genomics.net/case_studies/lambdaphage_demo.html</a>) from the website that accompanies the book Introduction to Computational Genomics: a case studies approach by Cristianini and Hahn (Cambridge University Press; www.computational-genomics.net/book/).”</p>
<div id="license" class="section level3">
<h3><span class="header-section-number">11.4.1</span> License</h3>
<p>The content in this book is licensed under a Creative Commons Attribution 3.0 License.</p>
</div>
</div>
<div id="exercises" class="section level2">
<h2><span class="header-section-number">11.5</span> Exercises</h2>
<p>Answer the following questions, using the R package. For each question, please record your answer, and what you typed into R to get this answer.</p>
<p>Model answers to the exercises are given in Answers to the exercises on DNA Sequence Statistics (2).</p>
<ol style="list-style-type: decimal">
<li>Draw a sliding window plot of GC content in the DEN-1 Dengue virus genome, using a window size of 200 nucleotides. Do you see any regions of unusual DNA content in the genome (eg. a high peak or low trough)?
Make a sketch of each plot that you draw. At what position (in base-pairs) in the genome is there the largest change in local GC content (approximate position is fine here)? Compare the sliding window plots of GC content created using window sizes of 200 and 2000 nucleotides. How does window size affect your ability to detect differences within the Dengue virus genome?</li>
<li>Draw a sliding window plot of GC content in the genome sequence for the bacterium Mycobacterium leprae strain TN (accession NC_002677) using a window size of 20000 nucleotides. Do you see any regions of unusual DNA content in the genome (eg. a high peak or low trough)?</li>
<li>Make a sketch of each plot that you drew for the previous quetion. Write down the approximate nucleotide position of the highest peak or lowest trough that you see. Why do you think a window size of 20000 nucleotides was chosen? What do you see if you use a much smaller windowsize (eg. 200 nucleotides) or a much larger windowsize (eg. 200,000 nucleotides)?</li>
<li>Advanced: Write a function to calculate the AT content of a DNA sequence (ie. the fraction of the nucleotides in the sequence that are As or Ts). What is the AT content of the Mycobacterium leprae TN genome? Hint: use the function count() to make a table containing the number of As, Gs, Ts and Cs in the sequence. Remember that function count() produces a table object, and you can access the elements of a table object using double square brackets. Do you notice a relationship between the AT content of the Mycobacterium leprae TN genome, and its GC content?</li>
<li>Advanced: Write a function to draw a sliding window plot of AT content. Use it to make a sliding window plot of AT content along the Mycobacterium leprae TN genome, using a windowsize of 20000 nucleotides. Do you notice any relationship between the sliding window plot of GC content along the Mycobacterium leprae genome, and the sliding window plot of AT content? Make a sketch of the plot that you draw.</li>
<li>Is the 3-nucleotide word GAC GC over-represented or under-represented in the Mycobacterium leprae TN genome sequence? What is the frequency of this word in the sequence? What is the expected frequency of this word in the sequence? What is the p (Rho) value for this word? How would you figure out whether there is already an R function to calculate p (Rho)? Is there one that you could use?</li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sequence-databases.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="alignments-in-r-not-updayed.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["compbio4all.pdf", "compbio4all.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
