<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 18 Rooted Neighbor-joing Phylogenetic trees | Computational Biology for All</title>
  <meta name="description" content="A bioinformatics and computational biology book" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 18 Rooted Neighbor-joing Phylogenetic trees | Computational Biology for All" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A bioinformatics and computational biology book" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 18 Rooted Neighbor-joing Phylogenetic trees | Computational Biology for All" />
  
  <meta name="twitter:description" content="A bioinformatics and computational biology book" />
  

<meta name="author" content="Avril Coglan and Nathan Brouwer" />


<meta name="date" content="2021-01-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="significance-of-orfs.html"/>
<link rel="next" href="todo.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Computational Biology for All</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#changes-made-to-original-source"><i class="fa fa-check"></i><b>1.1</b> Changes made to original source</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="downloading-r.html"><a href="downloading-r.html"><i class="fa fa-check"></i><b>2</b> Downloading R</a><ul>
<li class="chapter" data-level="2.1" data-path="downloading-r.html"><a href="downloading-r.html#introduction-to-r"><i class="fa fa-check"></i><b>2.1</b> Introduction to R</a></li>
<li class="chapter" data-level="2.2" data-path="downloading-r.html"><a href="downloading-r.html#installing-r"><i class="fa fa-check"></i><b>2.2</b> Installing R</a><ul>
<li class="chapter" data-level="2.2.1" data-path="downloading-r.html"><a href="downloading-r.html#installing-r-on-a-windows-pc"><i class="fa fa-check"></i><b>2.2.1</b> Installing R on a Windows PC</a></li>
<li class="chapter" data-level="2.2.2" data-path="downloading-r.html"><a href="downloading-r.html#how-to-install-r-on-non-windows-computers-eg.-macintosh-or-linux-computers"><i class="fa fa-check"></i><b>2.2.2</b> How to install R on non-Windows computers (eg. Macintosh or Linux computers)</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="downloading-r.html"><a href="downloading-r.html#starting-r"><i class="fa fa-check"></i><b>2.3</b> Starting R</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="downloading-rstudio.html"><a href="downloading-rstudio.html"><i class="fa fa-check"></i><b>3</b> Downloading RStudio</a><ul>
<li class="chapter" data-level="3.1" data-path="downloading-rstudio.html"><a href="downloading-rstudio.html#installing-rstudio"><i class="fa fa-check"></i><b>3.1</b> Installing RStudio</a></li>
<li class="chapter" data-level="3.2" data-path="downloading-rstudio.html"><a href="downloading-rstudio.html#getting-to-know-rstudio"><i class="fa fa-check"></i><b>3.2</b> Getting to know RStudio</a></li>
<li class="chapter" data-level="3.3" data-path="downloading-rstudio.html"><a href="downloading-rstudio.html#rstudio-versus-rstudio-cloud"><i class="fa fa-check"></i><b>3.3</b> RStudio versus RStudio Cloud</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="installing-r-packages.html"><a href="installing-r-packages.html"><i class="fa fa-check"></i><b>4</b> Installing R packages</a><ul>
<li class="chapter" data-level="4.1" data-path="installing-r-packages.html"><a href="installing-r-packages.html#downloading-packages-with-the-rstudio-gui"><i class="fa fa-check"></i><b>4.1</b> Downloading packages with the RStudio GUI</a></li>
<li class="chapter" data-level="4.2" data-path="installing-r-packages.html"><a href="installing-r-packages.html#downloading-packages-with-the-function-install.packages"><i class="fa fa-check"></i><b>4.2</b> Downloading packages with the function install.packages()</a></li>
<li class="chapter" data-level="4.3" data-path="installing-r-packages.html"><a href="installing-r-packages.html#using-packages-after-they-are-downloaded"><i class="fa fa-check"></i><b>4.3</b> Using packages after they are downloaded</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html"><i class="fa fa-check"></i><b>5</b> Installing Bioconductor</a><ul>
<li class="chapter" data-level="5.1" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html#bioconductor"><i class="fa fa-check"></i><b>5.1</b> Bioconductor</a></li>
<li class="chapter" data-level="5.2" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html#installing-biocmanager"><i class="fa fa-check"></i><b>5.2</b> Installing BiocManager</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html"><i class="fa fa-check"></i><b>6</b> Brief introduction to R</a><ul>
<li class="chapter" data-level="6.1" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#vocabulary"><i class="fa fa-check"></i><b>6.1</b> Vocabulary</a></li>
<li class="chapter" data-level="6.2" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#r-functions-etc"><i class="fa fa-check"></i><b>6.2</b> R functions etc</a></li>
<li class="chapter" data-level="6.3" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#a-brief-introduction-to-r"><i class="fa fa-check"></i><b>6.3</b> A brief introduction to R</a></li>
<li class="chapter" data-level="6.4" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#links-and-further-reading"><i class="fa fa-check"></i><b>6.4</b> Links and Further Reading</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html"><i class="fa fa-check"></i><b>7</b> Introduction to NCBI databases</a><ul>
<li class="chapter" data-level="7.1" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#to-do"><i class="fa fa-check"></i><b>7.1</b> To do</a></li>
<li class="chapter" data-level="7.2" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#the-ncbi-sequence-database"><i class="fa fa-check"></i><b>7.2</b> The NCBI Sequence Database</a></li>
<li class="chapter" data-level="7.3" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#the-ncbi-sub-databases"><i class="fa fa-check"></i><b>7.3</b> The NCBI Sub-Databases</a></li>
<li class="chapter" data-level="7.4" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#searching-for-an-accession-number-in-the-ncbi-database"><i class="fa fa-check"></i><b>7.4</b> Searching for an accession number in the NCBI database</a></li>
<li class="chapter" data-level="7.5" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#ncbi-sequence-format-ncbi-format"><i class="fa fa-check"></i><b>7.5</b> NCBI Sequence Format (NCBI Format)</a></li>
<li class="chapter" data-level="7.6" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#refseq"><i class="fa fa-check"></i><b>7.6</b> RefSeq</a></li>
<li class="chapter" data-level="7.7" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#querying-the-ncbi-database"><i class="fa fa-check"></i><b>7.7</b> Querying the NCBI Database</a></li>
<li class="chapter" data-level="7.8" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#querying-the-ncbi-database-via-the-ncbi-website-for-reference"><i class="fa fa-check"></i><b>7.8</b> Querying the NCBI Database via the NCBI Website (for reference)</a></li>
<li class="chapter" data-level="7.9" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#example-finding-the-sequences-published-in-nature-460352-358-for-reference"><i class="fa fa-check"></i><b>7.9</b> Example: finding the sequences published in Nature 460:352-358 (for reference)</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="downloading-ncbi-sequence-data-by-hand.html"><a href="downloading-ncbi-sequence-data-by-hand.html"><i class="fa fa-check"></i><b>8</b> Downloading NCBI sequence data by hand</a><ul>
<li class="chapter" data-level="8.1" data-path="downloading-ncbi-sequence-data-by-hand.html"><a href="downloading-ncbi-sequence-data-by-hand.html#retrieving-genome-sequence-data-via-the-ncbi-website"><i class="fa fa-check"></i><b>8.1</b> Retrieving genome sequence data via the NCBI website</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html"><i class="fa fa-check"></i><b>9</b> Downloading sequences from UniProt by hand</a><ul>
<li class="chapter" data-level="9.1" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#vocab"><i class="fa fa-check"></i><b>9.1</b> Vocab</a></li>
<li class="chapter" data-level="9.2" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#uniprot"><i class="fa fa-check"></i><b>9.2</b> UniProt</a></li>
<li class="chapter" data-level="9.3" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#viewing-the-uniprot-webpage-for-a-protein-sequence"><i class="fa fa-check"></i><b>9.3</b> Viewing the UniProt webpage for a protein sequence</a><ul>
<li class="chapter" data-level="9.3.1" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#protein-function"><i class="fa fa-check"></i><b>9.3.1</b> Protein function</a></li>
<li class="chapter" data-level="9.3.2" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#protein-sequence-and-size"><i class="fa fa-check"></i><b>9.3.2</b> Protein sequence and size</a></li>
<li class="chapter" data-level="9.3.3" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#other-information"><i class="fa fa-check"></i><b>9.3.3</b> Other information</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#retrieving-a-uniprot-protein-sequence-via-the-uniprot-website"><i class="fa fa-check"></i><b>9.4</b> Retrieving a UniProt protein sequence via the UniProt website</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="sequence-databases.html"><a href="sequence-databases.html"><i class="fa fa-check"></i><b>10</b> Sequence Databases</a><ul>
<li class="chapter" data-level="10.1" data-path="sequence-databases.html"><a href="sequence-databases.html#querying-the-ncbi-database-via-r"><i class="fa fa-check"></i><b>10.1</b> Querying the NCBI Database via R</a></li>
<li class="chapter" data-level="10.2" data-path="sequence-databases.html"><a href="sequence-databases.html#summary"><i class="fa fa-check"></i><b>10.2</b> Summary</a></li>
<li class="chapter" data-level="10.3" data-path="sequence-databases.html"><a href="sequence-databases.html#acknowledgements"><i class="fa fa-check"></i><b>10.3</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html"><i class="fa fa-check"></i><b>11</b> Local variation in GC content - NOT UPDATED</a><ul>
<li class="chapter" data-level="11.0.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#note-on-the-biology-in-this-section"><i class="fa fa-check"></i><b>11.0.1</b> Note on the biology in this section</a></li>
<li class="chapter" data-level="11.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#vocabulary-1"><i class="fa fa-check"></i><b>11.1</b> Vocabulary</a></li>
<li class="chapter" data-level="11.2" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#reading-sequence-data-with-rentrezentrez_fetch"><i class="fa fa-check"></i><b>11.2</b> Reading sequence data with rentrez::entrez_fetch</a></li>
<li class="chapter" data-level="11.3" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#local-variation-in-gc-content"><i class="fa fa-check"></i><b>11.3</b> Local variation in GC content</a></li>
<li class="chapter" data-level="11.4" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#acknowledgements-1"><i class="fa fa-check"></i><b>11.4</b> Acknowledgements</a><ul>
<li class="chapter" data-level="11.4.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#license"><i class="fa fa-check"></i><b>11.4.1</b> License</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#exercises"><i class="fa fa-check"></i><b>11.5</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html"><i class="fa fa-check"></i><b>12</b> Alignments in R - NOT UPDAYED</a><ul>
<li class="chapter" data-level="12.1" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#retrieving-a-uniprot-protein-sequence-using-seqinr"><i class="fa fa-check"></i><b>12.1</b> Retrieving a UniProt protein sequence using SeqinR</a></li>
<li class="chapter" data-level="12.2" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#comparing-two-sequences-using-a-dotplot"><i class="fa fa-check"></i><b>12.2</b> Comparing two sequences using a dotplot</a></li>
<li class="chapter" data-level="12.3" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#aligning-uniprot-sequences"><i class="fa fa-check"></i><b>12.3</b> Aligning UniProt sequences</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html"><i class="fa fa-check"></i><b>13</b> Multiple alignment with Clustal</a><ul>
<li class="chapter" data-level="13.1" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#installing-the-clustal-multiple-alignment-software"><i class="fa fa-check"></i><b>13.1</b> Installing the CLUSTAL multiple alignment software</a></li>
<li class="chapter" data-level="13.2" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#discarding-very-poorly-conserved-regions-from-an-alignment"><i class="fa fa-check"></i><b>13.2</b> Discarding very poorly conserved regions from an alignment</a></li>
<li class="chapter" data-level="13.3" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#calculating-genetic-distances-between-protein-sequences"><i class="fa fa-check"></i><b>13.3</b> Calculating genetic distances between protein sequences</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html"><i class="fa fa-check"></i><b>14</b> Unrooted Neighbor-joining Phylogenetic trees</a><ul>
<li class="chapter" data-level="14.1" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#building-an-unrooted-phylogenetic-tree-for-protein-sequences"><i class="fa fa-check"></i><b>14.1</b> Building an unrooted phylogenetic tree for protein sequences</a></li>
<li class="chapter" data-level="14.2" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#calculating-the-statistical-significance-of-a-pairwise-global-alignment"><i class="fa fa-check"></i><b>14.2</b> Calculating the statistical significance of a pairwise global alignment</a></li>
<li class="chapter" data-level="14.3" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#summary-1"><i class="fa fa-check"></i><b>14.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html"><i class="fa fa-check"></i><b>15</b> Retrieving multiple sequences in R</a><ul>
<li class="chapter" data-level="15.1" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html#prelminaries"><i class="fa fa-check"></i><b>15.1</b> Prelminaries</a></li>
<li class="chapter" data-level="15.2" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html#retrieving-a-list-of-sequences-from-uniprot"><i class="fa fa-check"></i><b>15.2</b> Retrieving a list of sequences from UniProt</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html"><i class="fa fa-check"></i><b>16</b> Computational gene finding</a><ul>
<li class="chapter" data-level="16.1" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#the-genetic-code"><i class="fa fa-check"></i><b>16.1</b> The genetic code</a></li>
<li class="chapter" data-level="16.2" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#reading-frames"><i class="fa fa-check"></i><b>16.2</b> Reading frames</a></li>
<li class="chapter" data-level="16.3" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#finding-open-reading-frames-on-the-forward-strand-of-a-dna-sequence"><i class="fa fa-check"></i><b>16.3</b> Finding open reading frames on the forward strand of a DNA sequence</a></li>
<li class="chapter" data-level="16.4" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#lengths-of-open-reading-frames"><i class="fa fa-check"></i><b>16.4</b> Lengths of open reading frames</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html"><i class="fa fa-check"></i><b>17</b> Significance of ORFs</a><ul>
<li class="chapter" data-level="17.1" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html#identifying-significant-open-reading-frames"><i class="fa fa-check"></i><b>17.1</b> Identifying significant open reading frames</a></li>
<li class="chapter" data-level="17.2" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html#summary-2"><i class="fa fa-check"></i><b>17.2</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="rooted-neighbor-joing-phylogenetic-trees.html"><a href="rooted-neighbor-joing-phylogenetic-trees.html"><i class="fa fa-check"></i><b>18</b> Rooted Neighbor-joing Phylogenetic trees</a><ul>
<li class="chapter" data-level="18.1" data-path="rooted-neighbor-joing-phylogenetic-trees.html"><a href="rooted-neighbor-joing-phylogenetic-trees.html#building-a-rooted-phylogenetic-tree-for-protein-sequences"><i class="fa fa-check"></i><b>18.1</b> Building a rooted phylogenetic tree for protein sequences</a></li>
<li class="chapter" data-level="18.2" data-path="rooted-neighbor-joing-phylogenetic-trees.html"><a href="rooted-neighbor-joing-phylogenetic-trees.html#building-a-phylogenetic-tree-for-dna-or-mrna-sequences"><i class="fa fa-check"></i><b>18.2</b> Building a phylogenetic tree for DNA or mRNA sequences</a></li>
<li class="chapter" data-level="18.3" data-path="rooted-neighbor-joing-phylogenetic-trees.html"><a href="rooted-neighbor-joing-phylogenetic-trees.html#summary-3"><i class="fa fa-check"></i><b>18.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="todo.html"><a href="todo.html"><i class="fa fa-check"></i><b>19</b> TODO</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://https://github.com/brouwern/CB4ALL" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Computational Biology for All</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rooted-neighbor-joing-phylogenetic-trees" class="section level1">
<h1><span class="header-section-number">Chapter 18</span> Rooted Neighbor-joing Phylogenetic trees</h1>
<p>By: Avril Coghlan, adapted by Nathan Brouwer</p>
<div id="building-a-rooted-phylogenetic-tree-for-protein-sequences" class="section level2">
<h2><span class="header-section-number">18.1</span> Building a rooted phylogenetic tree for protein sequences</h2>
<p>In order to convert the unrooted tree into a rooted tree, we need to add an outgroup sequence. Normally, the outgroup sequence is a sequence that we know from some prior knowledge to be more distantly related to the other sequences under study than they are to each other.</p>
<p>For example, the protein Fox-1 is involved in determining the sex (gender) of an embryo in the nematode worm Caenorhabditis elegans (UniProt accession Q10572). Related proteins are found in other nematodes, including Caenorhabditis remanei (UniProt E3M2K8), Caenorhabditis briggsae (A8WS01), Loa loa (E1FUV2), and Brugia malayi (UniProt A8NSK3).</p>
<p>Note that Caenorhabditis elegans is a model organism commonly studied in molecular biology. The nematodes Loa loa, and Brugia malayi are parasitic nematodes that cause filariasis, which is classified by the WHO as a neglected tropical disease.</p>
<p>The UniProt database contains a distantly related sequence from the fruitfly Drosophila melanogaster (UniProt accession Q9VT99). If we were to build a phylogenetic tree of the nematode worm Fox-1 homologues, the distantly related sequence from fruitfly would probably be a good choice of outgroup, since the protein is from a different animal group (insects) than the nematode worms. Thus, it is likely that the fruitfly protein is more distantly related to all the nematode proteins than they are to each other.</p>
<p>To retrieve the sequences from UniProt we can use the “retrieveseqs()” function (see above):</p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="rooted-neighbor-joing-phylogenetic-trees.html#cb160-1"></a>seqnames &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Q10572&quot;</span>,<span class="st">&quot;E3M2K8&quot;</span>,<span class="st">&quot;Q8WS01&quot;</span>,<span class="st">&quot;E1FUV2&quot;</span>,<span class="st">&quot;A8NSK3&quot;</span>,<span class="st">&quot;Q9VT99&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="rooted-neighbor-joing-phylogenetic-trees.html#cb161-1"></a>seqs &lt;-<span class="st"> </span><span class="kw">retrieveseqs</span>(seqnames,<span class="st">&quot;swissprot&quot;</span>)</span></code></pre></div>
<p>We can then write out the sequences to a FASTA file:</p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="rooted-neighbor-joing-phylogenetic-trees.html#cb162-1"></a><span class="kw">write.fasta</span>(seqs, seqnames, <span class="dt">file=</span><span class="st">&quot;fox1.fasta&quot;</span>)</span></code></pre></div>
<p>We can then use CLUSTAL to create a PHYLIP-format alignment of the sequences, and store it in the alignment file “fox1.phy”. This picture shows part of the alignment (the alignment is quite long, so not all of it is shown):</p>
<p>image10</p>
<p>We can then read the alignment into R:</p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="rooted-neighbor-joing-phylogenetic-trees.html#cb163-1"></a>fox1aln  &lt;-<span class="st"> </span><span class="kw">read.alignment</span>(<span class="dt">file =</span> <span class="st">&quot;fox1.phy&quot;</span>, <span class="dt">format =</span> <span class="st">&quot;phylip&quot;</span>)</span></code></pre></div>
<p>The next step is to build a phylogenetic tree of the proteins, which again we can do using the neighbour-joining algorithm.</p>
<p>This time we have an outgroup in our set of sequences, so we can build a rooted tree. The function “rootedNJtree()” can be used to build a rooted tree. It returns the phylogenetic tree, and also makes a picture of the tree:</p>
<p>The function takes the alignment and the name of the outgroup as its inputs. For example, to use it to make a phylogenetic tree of the C. elegans Fox-1 protein and its homologues, using the fruitfly protein (UniProt Q9VT99) as the outgroup, we type:</p>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="rooted-neighbor-joing-phylogenetic-trees.html#cb164-1"></a>fox1alntree &lt;-<span class="st"> </span><span class="kw">rootedNJtree</span>(fox1aln, <span class="st">&quot;Q9VT99&quot;</span>,<span class="dt">type=</span><span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<p>Here we can see that E3M2K8 (C. remanei Fox-1 homologue) and Q10572 (C. elegans Fox-1) have been grouped together with bootstrap 100%, and A8NSK3 (Brugia malayi Fox-1 homologue) and E1FUV2 (Loa loa Fox-1 homologue) have been grouped together with bootstrap 100%. These four proteins have also been grouped together in a larger clade with bootstrap 100%.</p>
<p>Compared to these four proteins, the Q8WS01 (C. briggsae Fox-1 homologue) and Q9VT99 (fruitfly outgroup) seem to be relatively distantly related.</p>
<p>As this is a rooted tree, we know the direction that evolutionary time ran. Say we call the ancestor of the four sequences (E3M2K8, Q10572, A8NSK3, E1FUV2) ancestor1, the ancestor of the two sequences (E3M2K8, Q10572) ancestor2, and the ancestor of the two sequences (A8NSK3, E1FUV2) ancestor3.</p>
<p>Because it is a rooted tree, we know that time ran from left to right along the branches of the tree, so that ancestor1 was the ancestor of ancestor2, and ancestor1 was also the ancestor of ancestor3. In other words, ancestor1 lived before ancestor2 or ancestor3; ancestor2 and ancestor3 were descendants of ancestor1.</p>
<p>Another way of saying this is that E3M2K8 and Q10572 shared a common ancestor with each other more recently than they did with A8NSK3 and E1FUV2.</p>
<p>The lengths of branches in this tree are proportional to the amount of evolutionary change (estimated number of mutations) that occurred along the branches. The branches leading back from E3M2K8 and Q10572 to their last common ancestor are slightly longer than the branches leading back from A8NSK3 and E1FUV2 to their last common ancestor.</p>
<p>This indicates that there has been more evolutionary change in E3M2K8 (C. remanei Fox-1 homologue) and Q10572 (C. elegans Fox-1) proteins since they diverged, than there has been in A8NSK3 (Brugia malayi Fox-1 homologue) and E1FUV2 (Loa loa Fox-1 homologue) since they diverged.</p>
</div>
<div id="building-a-phylogenetic-tree-for-dna-or-mrna-sequences" class="section level2">
<h2><span class="header-section-number">18.2</span> Building a phylogenetic tree for DNA or mRNA sequences</h2>
<p>In the example above, a phylogenetic tree was built for protein sequences. The genomes of distantly related organisms such as vertebrates will have accumulated many mutations since they diverged. Sometimes, so many mutations have occurred since the organisms diverged that their DNA sequences are hard to align correctly and it is also hard to accurately estimate evolutionary distances from alignments of those DNA sequences.</p>
<p>In contrast, as many mutations at the DNA level are synonymous at the protein level, protein sequences diverge at a slower rate than DNA sequences. This is why for reasonably distantly related organisms such as vertebrates, it is usually preferable to use protein sequences for phylogenetic analyses.</p>
<p>If you are studying closely related organisms such as primates, few mutations will have occurred since they diverged. As a result, if you use protein sequences for a phylogenetic analysis, there may be too few amino acid substitutions to provide enough ‘signal’ to use for the phylogenetic analysis. Therefore, it is often preferable to use DNA sequences for a phylogenetic analysis of closely related organisms such as primates.</p>
<p>We can use the functions unrootedNJtree() and rootedNJtree() described above to build unrooted or rooted neighbour-joining phylogenetic trees based on an alignment of DNA or mRNA sequences. In this case, we need to use “type=DNA” as an argument in these functions, to tell them that we are making a tree of nucleotide sequences, not protein sequences.</p>
<p>For example, to build an unrooted phylogenetic tree based on the alignment of the virus phosphoprotein mRNA sequences, we type in R:</p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="rooted-neighbor-joing-phylogenetic-trees.html#cb165-1"></a>virusmRNAaln  &lt;-<span class="st"> </span><span class="kw">read.alignment</span>(<span class="dt">file =</span> <span class="st">&quot;virusmRNA.phy&quot;</span>, <span class="dt">format =</span> <span class="st">&quot;phylip&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="rooted-neighbor-joing-phylogenetic-trees.html#cb166-1"></a>virusmRNAalntree &lt;-<span class="st"> </span><span class="kw">unrootedNJtree</span>(virusmRNAaln, <span class="dt">type=</span><span class="st">&quot;DNA&quot;</span>)</span></code></pre></div>
<p>Saving a phylogenetic tree as a Newick-format tree file
A commonly used format for representing phylogenetic trees is the Newick format. Once you have built a phylogenetic tree using R, it is convenient to store it as a Newick-format tree file. This can be done using the “write.tree()” function in the Ape R package.</p>
<p>For example, to save the unrooted phylogenetic tree of virus phosphoprotein mRNA sequences as a Newick-format tree file called “virusmRNA.tre”, we type:</p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="rooted-neighbor-joing-phylogenetic-trees.html#cb167-1"></a><span class="kw">write.tree</span>(virusmRNAalntree, <span class="st">&quot;virusmRNA.tre&quot;</span>)</span></code></pre></div>
<p>The Newick-format file “virusmRNA.tre” should now appear in your “My Documents” folder.</p>
</div>
<div id="summary-3" class="section level2">
<h2><span class="header-section-number">18.3</span> Summary</h2>
<p>In this practical, you have learnt the following R functions that belong to the bioinformatics packages:</p>
<p>read.alignment() from the SeqinR package for reading in a multiple alignment
dist.alignment() from the SeqinR package for calculating genetic distances between protein sequences
dist.dna() from the Ape package for calculating genetic distances between DNA or mRNA sequences
Links and Further Reading
Some links are included here for further reading.</p>
<p>For background reading on phylogenetic trees, it is recommended to read Chapter 7 of Introduction to Computational Genomics: a case studies approach by Cristianini and Hahn (Cambridge University Press; www.computational-genomics.net/book/).</p>
<p>Another more in-depth (but very accesssible) book on phylogenetics is Molecular Evolution: A Phylogenetic Approach by Roderic DM Page and Edward C Holmes.</p>
<p>For more in-depth information and more examples on using the SeqinR package for sequence analysis, look at the SeqinR documentation, <a href="http://pbil.univ-lyon1.fr/software/seqinr/doc.php?lang=eng" class="uri">http://pbil.univ-lyon1.fr/software/seqinr/doc.php?lang=eng</a>.</p>
<p>For more in-depth information and more examples on the Ape package for phylogenetic analysis, look at the Ape documentation, ape.mpl.ird.fr/.</p>
<p>If you are using the Ape package for a phylogenetic analysis project, it would be worthwhile to obtain a copy of the book Analysis of Phylogenetics and Evolution with R by Emmanuel Paradis, published by Springer, which has many nice examples of using R for phylogenetic analyses.</p>
<p>For a more in-depth introduction to R, a good online tutorial is available on the “Kickstarting R” website, cran.r-project.org/doc/contrib/Lemon-kickstart.</p>
<p>There is another nice (slightly more in-depth) tutorial to R available on the “Introduction to R” website, cran.r-project.org/doc/manuals/R-intro.html.</p>
<p>Acknowledgements
Many of the ideas for the examples and exercises for this practical were inspired by the Matlab case study on SARS (www.computational-genomics.net/case_studies/sars_demo.html) from the website that accompanies the book Introduction to Computational Genomics: a case studies approach by Cristianini and Hahn (Cambridge University Press; www.computational-genomics.net/book/).</p>
<p>Thank you to Jean Lobry and Simon Penel for helpful advice on using the SeqinR package.</p>
<p>Thank you to Emmanuel Paradis and François Michonneau for help in using the Ape package.</p>
<p>Thank you also to Klaus Schliep for helpful comments.</p>
<p>Contact
I will be grateful if you will send me (Avril Coghlan) corrections or suggestions for improvements to my email address <a href="mailto:alc@sanger.ac.uk" class="email">alc@sanger.ac.uk</a></p>
<p>License
The content in this book is licensed under a Creative Commons Attribution 3.0 License.</p>
<p>Exercises
Answer the following questions, using the R package. For each question, please record your answer, and what you typed into R to get this answer.</p>
<p>Model answers to the exercises are given in Answers to the exercises on Multiple Alignment and Phylogenetic Trees.</p>
<ol style="list-style-type: decimal">
<li>Calculate the genetic distances between the following NS1 proteins from different Dengue virus strains: Dengue virus 1 NS1 protein (Uniprot Q9YRR4), Dengue virus 2 NS1 protein (UniProt Q9YP96), Dengue virus 3 NS1 protein (UniProt B0LSS3), and Dengue virus 4 NS1 protein (UniProt Q6TFL5). Which are the most closely related proteins, and which are the least closely related, based on the genetic distances?
Note: Dengue virus causes Dengue fever, which is classified by the WHO as a neglected tropical disease. There are four main types of Dengue virus, Dengue virus 1, Dengue virus 2, Dengue virus 3, and Dengue virus 4.</li>
<li>Build an unrooted phylogenetic tree of the NS1 proteins from Dengue virus 1, Dengue virus 2, Dengue virus 3 and Dengue virus 4, using the neighbour-joining algorithm. Which are the most closely related proteins, based on the tree? Based on the bootstrap values in the tree, how confident are you of this?</li>
<li>Build an unrooted phylogenetic tree of the NS1 proteins from Dengue viruses 1-4, based on a filtered alignment of the four proteins (keeping alignment columns in which at least 30% of letters are not gaps, and in which at least 30% of pairs of letters are identical). Does this differ from the tree based on the unfiltered alignment (in Q2)? Can you explain why?</li>
<li>The Zika virus is related to Dengue viruses, but is not a Dengue virus, and so therefore can be used as an outgroup in phylogenetic trees of Dengue virus sequences. UniProt accession Q32ZE1 consists of a sequence with similarity to the Dengue NS1 protein, so seems to be a related protein from Zika virus. Build a rooted phylogenetic tree of the Dengue NS1 proteins based on a filtered alignment (keeping alignment columns in which at least 30% of letters are not gaps, and in which at least 30% of pairs of letters are identical), using the Zika virus protein as the outgroup. Which are the most closely related Dengue virus proteins, based on the tree? What extra information does this tree tell you, compared to the unrooted tree in Q2?</li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="significance-of-orfs.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="todo.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["compbio4all.pdf", "compbio4all.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
