<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 14 Unrooted Neighbor-joining Phylogenetic trees | Computational Biology for All!</title>
  <meta name="description" content="A bioinformatics and computational biology book" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 14 Unrooted Neighbor-joining Phylogenetic trees | Computational Biology for All!" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A bioinformatics and computational biology book" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 14 Unrooted Neighbor-joining Phylogenetic trees | Computational Biology for All!" />
  
  <meta name="twitter:description" content="A bioinformatics and computational biology book" />
  

<meta name="author" content="Avril Coglan and Nathan Brouwer" />


<meta name="date" content="2021-01-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="multiple-alignment-with-clustal.html"/>
<link rel="next" href="retrieving-multiple-sequences-in-r.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Computational Biology for All</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="downloading-r.html"><a href="downloading-r.html"><i class="fa fa-check"></i><b>2</b> Downloading R</a><ul>
<li class="chapter" data-level="2.1" data-path="downloading-r.html"><a href="downloading-r.html#introduction-to-r"><i class="fa fa-check"></i><b>2.1</b> Introduction to R</a></li>
<li class="chapter" data-level="2.2" data-path="downloading-r.html"><a href="downloading-r.html#installing-r"><i class="fa fa-check"></i><b>2.2</b> Installing R</a><ul>
<li class="chapter" data-level="2.2.1" data-path="downloading-r.html"><a href="downloading-r.html#installing-r-on-a-windows-pc"><i class="fa fa-check"></i><b>2.2.1</b> Installing R on a Windows PC</a></li>
<li class="chapter" data-level="2.2.2" data-path="downloading-r.html"><a href="downloading-r.html#how-to-install-r-on-non-windows-computers-eg.-macintosh-or-linux-computers"><i class="fa fa-check"></i><b>2.2.2</b> How to install R on non-Windows computers (eg. Macintosh or Linux computers)</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="downloading-r.html"><a href="downloading-r.html#starting-r"><i class="fa fa-check"></i><b>2.3</b> Starting R</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="downloading-the-rstudio.html"><a href="downloading-the-rstudio.html"><i class="fa fa-check"></i><b>3</b> Downloading the RStudio</a><ul>
<li class="chapter" data-level="3.1" data-path="downloading-the-rstudio.html"><a href="downloading-the-rstudio.html#installing-the-rstudio-ide"><i class="fa fa-check"></i><b>3.1</b> Installing the RStudio IDE</a></li>
<li class="chapter" data-level="3.2" data-path="downloading-the-rstudio.html"><a href="downloading-the-rstudio.html#getting-to-know-rstudio"><i class="fa fa-check"></i><b>3.2</b> Getting to know RStudio</a></li>
<li class="chapter" data-level="3.3" data-path="downloading-the-rstudio.html"><a href="downloading-the-rstudio.html#rstudio-versus-rstudio-cloud"><i class="fa fa-check"></i><b>3.3</b> RStudio versus RStudio Cloud</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="installing-r-packages.html"><a href="installing-r-packages.html"><i class="fa fa-check"></i><b>4</b> Installing R packages</a><ul>
<li class="chapter" data-level="4.1" data-path="installing-r-packages.html"><a href="installing-r-packages.html#downloading-packages-with-the-rstudio-ide"><i class="fa fa-check"></i><b>4.1</b> Downloading packages with the RStudio IDE</a></li>
<li class="chapter" data-level="4.2" data-path="installing-r-packages.html"><a href="installing-r-packages.html#downloading-packages-with-the-function-install.packages"><i class="fa fa-check"></i><b>4.2</b> Downloading packages with the function install.packages()</a></li>
<li class="chapter" data-level="4.3" data-path="installing-r-packages.html"><a href="installing-r-packages.html#using-packages-after-they-are-downloaded"><i class="fa fa-check"></i><b>4.3</b> Using packages after they are downloaded</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html"><i class="fa fa-check"></i><b>5</b> Installing Bioconductor</a><ul>
<li class="chapter" data-level="5.1" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html#bioconductor"><i class="fa fa-check"></i><b>5.1</b> Bioconductor</a></li>
<li class="chapter" data-level="5.2" data-path="installing-bioconductor.html"><a href="installing-bioconductor.html#installing-biocmanager"><i class="fa fa-check"></i><b>5.2</b> Installing BiocManager</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html"><i class="fa fa-check"></i><b>6</b> Brief introduction to R</a><ul>
<li class="chapter" data-level="6.1" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#vocabulary"><i class="fa fa-check"></i><b>6.1</b> Vocabulary</a></li>
<li class="chapter" data-level="6.2" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#r-functions-etc"><i class="fa fa-check"></i><b>6.2</b> R functions etc</a></li>
<li class="chapter" data-level="6.3" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#a-brief-introduction-to-r"><i class="fa fa-check"></i><b>6.3</b> A brief introduction to R</a></li>
<li class="chapter" data-level="6.4" data-path="brief-introduction-to-r.html"><a href="brief-introduction-to-r.html#links-and-further-reading"><i class="fa fa-check"></i><b>6.4</b> Links and Further Reading</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="downloading-ncbi-sequence-data-by-hand.html"><a href="downloading-ncbi-sequence-data-by-hand.html"><i class="fa fa-check"></i><b>7</b> Downloading NCBI sequence data by hand</a><ul>
<li class="chapter" data-level="7.1" data-path="downloading-ncbi-sequence-data-by-hand.html"><a href="downloading-ncbi-sequence-data-by-hand.html#retrieving-genome-sequence-data-via-the-ncbi-website"><i class="fa fa-check"></i><b>7.1</b> Retrieving genome sequence data via the NCBI website</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html"><i class="fa fa-check"></i><b>8</b> Introduction to NCBI databases</a><ul>
<li class="chapter" data-level="8.1" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#to-do"><i class="fa fa-check"></i><b>8.1</b> To do</a></li>
<li class="chapter" data-level="8.2" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#the-ncbi-sequence-database"><i class="fa fa-check"></i><b>8.2</b> The NCBI Sequence Database</a></li>
<li class="chapter" data-level="8.3" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#the-ncbi-sub-databases"><i class="fa fa-check"></i><b>8.3</b> The NCBI Sub-Databases</a></li>
<li class="chapter" data-level="8.4" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#searching-for-an-accession-number-in-the-ncbi-database"><i class="fa fa-check"></i><b>8.4</b> Searching for an accession number in the NCBI database</a></li>
<li class="chapter" data-level="8.5" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#ncbi-sequence-format-ncbi-format"><i class="fa fa-check"></i><b>8.5</b> NCBI Sequence Format (NCBI Format)</a></li>
<li class="chapter" data-level="8.6" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#refseq"><i class="fa fa-check"></i><b>8.6</b> RefSeq</a></li>
<li class="chapter" data-level="8.7" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#querying-the-ncbi-database"><i class="fa fa-check"></i><b>8.7</b> Querying the NCBI Database</a></li>
<li class="chapter" data-level="8.8" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#querying-the-ncbi-database-via-the-ncbi-website-for-reference"><i class="fa fa-check"></i><b>8.8</b> Querying the NCBI Database via the NCBI Website (for reference)</a></li>
<li class="chapter" data-level="8.9" data-path="introduction-to-ncbi-databases.html"><a href="introduction-to-ncbi-databases.html#example-finding-the-sequences-published-in-nature-460352-358-for-reference"><i class="fa fa-check"></i><b>8.9</b> Example: finding the sequences published in Nature 460:352-358 (for reference)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html"><i class="fa fa-check"></i><b>9</b> Downloading sequences from UniProt by hand</a><ul>
<li class="chapter" data-level="9.1" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#vocab"><i class="fa fa-check"></i><b>9.1</b> Vocab</a></li>
<li class="chapter" data-level="9.2" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#uniprot"><i class="fa fa-check"></i><b>9.2</b> UniProt</a></li>
<li class="chapter" data-level="9.3" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#viewing-the-uniprot-webpage-for-a-protein-sequence"><i class="fa fa-check"></i><b>9.3</b> Viewing the UniProt webpage for a protein sequence</a><ul>
<li class="chapter" data-level="9.3.1" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#protein-function"><i class="fa fa-check"></i><b>9.3.1</b> Protein function</a></li>
<li class="chapter" data-level="9.3.2" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#protein-sequence-and-size"><i class="fa fa-check"></i><b>9.3.2</b> Protein sequence and size</a></li>
<li class="chapter" data-level="9.3.3" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#other-information"><i class="fa fa-check"></i><b>9.3.3</b> Other information</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="downloading-sequences-from-uniprot-by-hand.html"><a href="downloading-sequences-from-uniprot-by-hand.html#retrieving-a-uniprot-protein-sequence-via-the-uniprot-website"><i class="fa fa-check"></i><b>9.4</b> Retrieving a UniProt protein sequence via the UniProt website</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="sequence-databases.html"><a href="sequence-databases.html"><i class="fa fa-check"></i><b>10</b> Sequence Databases</a><ul>
<li class="chapter" data-level="10.1" data-path="sequence-databases.html"><a href="sequence-databases.html#querying-the-ncbi-database-via-r"><i class="fa fa-check"></i><b>10.1</b> Querying the NCBI Database via R</a></li>
<li class="chapter" data-level="10.2" data-path="sequence-databases.html"><a href="sequence-databases.html#summary"><i class="fa fa-check"></i><b>10.2</b> Summary</a></li>
<li class="chapter" data-level="10.3" data-path="sequence-databases.html"><a href="sequence-databases.html#acknowledgements"><i class="fa fa-check"></i><b>10.3</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html"><i class="fa fa-check"></i><b>11</b> Local variation in GC content - NOT UPDATED</a><ul>
<li class="chapter" data-level="11.0.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#note-on-the-biology-in-this-section"><i class="fa fa-check"></i><b>11.0.1</b> Note on the biology in this section</a></li>
<li class="chapter" data-level="11.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#vocabulary-1"><i class="fa fa-check"></i><b>11.1</b> Vocabulary</a></li>
<li class="chapter" data-level="11.2" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#reading-sequence-data-with-rentrezentrez_fetch"><i class="fa fa-check"></i><b>11.2</b> Reading sequence data with rentrez::entrez_fetch</a></li>
<li class="chapter" data-level="11.3" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#local-variation-in-gc-content"><i class="fa fa-check"></i><b>11.3</b> Local variation in GC content</a></li>
<li class="chapter" data-level="11.4" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#a-sliding-window-analysis-of-gc-content"><i class="fa fa-check"></i><b>11.4</b> A sliding window analysis of GC content</a></li>
<li class="chapter" data-level="11.5" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#acknowledgements-1"><i class="fa fa-check"></i><b>11.5</b> Acknowledgements</a><ul>
<li class="chapter" data-level="11.5.1" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#license"><i class="fa fa-check"></i><b>11.5.1</b> License</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="local-variation-in-gc-content-not-updated.html"><a href="local-variation-in-gc-content-not-updated.html#exercises"><i class="fa fa-check"></i><b>11.6</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html"><i class="fa fa-check"></i><b>12</b> Alignments in R - NOT UPDAYED</a><ul>
<li class="chapter" data-level="12.1" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#retrieving-a-uniprot-protein-sequence-using-seqinr"><i class="fa fa-check"></i><b>12.1</b> Retrieving a UniProt protein sequence using SeqinR</a></li>
<li class="chapter" data-level="12.2" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#comparing-two-sequences-using-a-dotplot"><i class="fa fa-check"></i><b>12.2</b> Comparing two sequences using a dotplot</a></li>
<li class="chapter" data-level="12.3" data-path="alignments-in-r-not-updayed.html"><a href="alignments-in-r-not-updayed.html#aligning-uniprot-sequences"><i class="fa fa-check"></i><b>12.3</b> Aligning UniProt sequences</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html"><i class="fa fa-check"></i><b>13</b> Multiple alignment with Clustal</a><ul>
<li class="chapter" data-level="13.1" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#installing-the-clustal-multiple-alignment-software"><i class="fa fa-check"></i><b>13.1</b> Installing the CLUSTAL multiple alignment software</a></li>
<li class="chapter" data-level="13.2" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#discarding-very-poorly-conserved-regions-from-an-alignment"><i class="fa fa-check"></i><b>13.2</b> Discarding very poorly conserved regions from an alignment</a></li>
<li class="chapter" data-level="13.3" data-path="multiple-alignment-with-clustal.html"><a href="multiple-alignment-with-clustal.html#calculating-genetic-distances-between-protein-sequences"><i class="fa fa-check"></i><b>13.3</b> Calculating genetic distances between protein sequences</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html"><i class="fa fa-check"></i><b>14</b> Unrooted Neighbor-joining Phylogenetic trees</a><ul>
<li class="chapter" data-level="14.1" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#building-an-unrooted-phylogenetic-tree-for-protein-sequences"><i class="fa fa-check"></i><b>14.1</b> Building an unrooted phylogenetic tree for protein sequences</a></li>
<li class="chapter" data-level="14.2" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#testing-the-significance-of-an-alignment"><i class="fa fa-check"></i><b>14.2</b> Testing the significance of an alignment</a></li>
<li class="chapter" data-level="14.3" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#calculating-the-statistical-significance-of-a-pairwise-global-alignment"><i class="fa fa-check"></i><b>14.3</b> Calculating the statistical significance of a pairwise global alignment</a></li>
<li class="chapter" data-level="14.4" data-path="unrooted-neighbor-joining-phylogenetic-trees.html"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#summary-1"><i class="fa fa-check"></i><b>14.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html"><i class="fa fa-check"></i><b>15</b> Retrieving multiple sequences in R</a><ul>
<li class="chapter" data-level="15.1" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html#prelminaries"><i class="fa fa-check"></i><b>15.1</b> Prelminaries</a></li>
<li class="chapter" data-level="15.2" data-path="retrieving-multiple-sequences-in-r.html"><a href="retrieving-multiple-sequences-in-r.html#retrieving-a-list-of-sequences-from-uniprot"><i class="fa fa-check"></i><b>15.2</b> Retrieving a list of sequences from UniProt</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html"><i class="fa fa-check"></i><b>16</b> Computational gene finding</a><ul>
<li class="chapter" data-level="16.1" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#the-genetic-code"><i class="fa fa-check"></i><b>16.1</b> The genetic code</a></li>
<li class="chapter" data-level="16.2" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#reading-frames"><i class="fa fa-check"></i><b>16.2</b> Reading frames</a></li>
<li class="chapter" data-level="16.3" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#finding-open-reading-frames-on-the-forward-strand-of-a-dna-sequence"><i class="fa fa-check"></i><b>16.3</b> Finding open reading frames on the forward strand of a DNA sequence</a></li>
<li class="chapter" data-level="16.4" data-path="computational-gene-finding.html"><a href="computational-gene-finding.html#lengths-of-open-reading-frames"><i class="fa fa-check"></i><b>16.4</b> Lengths of open reading frames</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html"><i class="fa fa-check"></i><b>17</b> Significance of ORFs</a><ul>
<li class="chapter" data-level="17.1" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html#identifying-significant-open-reading-frames"><i class="fa fa-check"></i><b>17.1</b> Identifying significant open reading frames</a></li>
<li class="chapter" data-level="17.2" data-path="significance-of-orfs.html"><a href="significance-of-orfs.html#summary-2"><i class="fa fa-check"></i><b>17.2</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="rooted-neighbor-joining-phylogenetic-trees.html"><a href="rooted-neighbor-joining-phylogenetic-trees.html"><i class="fa fa-check"></i><b>18</b> Rooted Neighbor-joining Phylogenetic trees</a><ul>
<li class="chapter" data-level="18.1" data-path="rooted-neighbor-joining-phylogenetic-trees.html"><a href="rooted-neighbor-joining-phylogenetic-trees.html#building-a-rooted-phylogenetic-tree-for-protein-sequences"><i class="fa fa-check"></i><b>18.1</b> Building a rooted phylogenetic tree for protein sequences</a></li>
<li class="chapter" data-level="18.2" data-path="rooted-neighbor-joining-phylogenetic-trees.html"><a href="rooted-neighbor-joining-phylogenetic-trees.html#building-a-phylogenetic-tree-for-dna-or-mrna-sequences"><i class="fa fa-check"></i><b>18.2</b> Building a phylogenetic tree for DNA or mRNA sequences</a></li>
<li class="chapter" data-level="18.3" data-path="rooted-neighbor-joining-phylogenetic-trees.html"><a href="rooted-neighbor-joining-phylogenetic-trees.html#summary-3"><i class="fa fa-check"></i><b>18.3</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="todo.html"><a href="todo.html"><i class="fa fa-check"></i><b>19</b> TODO</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://https://github.com/brouwern/CB4ALL" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Computational Biology for All!</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="unrooted-neighbor-joining-phylogenetic-trees" class="section level1">
<h1><span class="header-section-number">Chapter 14</span> Unrooted Neighbor-joining Phylogenetic trees</h1>
<p><strong>By</strong>: Avril Coghlan. Adapted and edited by Nathan Brouwer.</p>
<div id="building-an-unrooted-phylogenetic-tree-for-protein-sequences" class="section level2">
<h2><span class="header-section-number">14.1</span> Building an unrooted phylogenetic tree for protein sequences</h2>
<p>Once we have a distance matrix that gives the pairwise distances between all our protein sequences, we can build a phylogenetic tree based on that distance matrix. One method for using this is the neighbour-joining algorithm.</p>
<p>You can build a phylogenetic tree using the neighbour-joining algorithm with the the Ape R package. First you will need to install the “ape” package (see instructions on how to install R packages).</p>
<p>The following R function “unrootedNJtree()” builds a phylogenetic tree based on an alignment of sequences, using the neighbour-joining algorithm, using functions from the “ape” package.</p>
<p>The “unrootedNJtree()” function takes an alignment of sequences its input, calculates pairwise distances between the sequences based on the alignment, and then builds a phylogenetic tree based on the pairwise distances. It returns the phylogenetic tree, and also makes a picture of that tree:</p>
<p>To use the function to make a phylogenetic tree, you must first copy and paste the function into R. You can then use it to make a tree, for example of the virus phosphoproteins, based on the sequence alignment:</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#cb98-1"></a>virusalntree &lt;-<span class="st"> </span><span class="kw">unrootedNJtree</span>(virusaln,<span class="dt">type=</span><span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<p>Note that you need to specify that the type of sequences that you are using are protein sequences when you use unrootedNJtree(), by setting “type=protein”.</p>
<p>We can see that Q5VKP1 (Western Caucasian bat virus phosphoprotein) and P06747 (rabies virus phosphoprotein) have been grouped together in the tree, and that O56773 (Lagos bat virus phosphoprotein) and P0C569 (Mokola virus phosphoprotein) are grouped together in the tree.</p>
<p>This is consistent with what we saw above in the genetic distance matrix, which showed that the genetic distance between Lagos bat virus phosphoprotein (O56773) and Mokola virus phosphoprotein (P0C569) is relatively small.</p>
<p>The numbers in blue boxes are bootstrap values for the nodes in the tree.</p>
<p>A bootstrap value for a particular node in the tree gives an idea of the confidence that we have in the clade (group) defined by that node in the tree. If a node has a high bootstrap value (near 100%) then we are very confident that the clade defined by the node is correct, while if it has a low bootstrap value (near 0%) then we are not so confident.</p>
<p>Note that the fact that a bootstrap value for a node is high does not necessarily guarantee that the clade defined by the node is correct, but just tells us that it is quite likely that it is correct.</p>
<p>The bootstrap values are calculated by making many (for example, 100) random “resamples” of the alignment that the phylogenetic tree was based upon. Each “resample” of the alignment consists of a certain number x (eg. 200) of randomly sampled columns from the alignment. Each “resample” of the alignment (eg. 200 randomly sampled columns) forms a sort of fake alignment of its own, and a phylogenetic tree can be based upon the “resample”. We can make 100 random resamples of the alignment, and build 100 phylogenetic trees based on the 100 resamples. These 100 trees are known as the “bootstrap trees”. For each clade (grouping) that we see in our original phylogenetic tree, we can count in how many of the 100 bootstrap trees it appears. This is known as the “bootstrap value” for the clade in our original phylogenetic tree.</p>
<p>For example, if we calculate 100 random resamples of the virus phosphoprotein alignment, and build 100 phylogenetic trees based on these resamples, we can calculate the bootstrap values for each clade in the virus phosphoprotein phylogenetic tree.</p>
<p>In this case, the bootstrap value for the node defining the clade containing Q5VKP1 (Western Caucasian bat virus phosphoprotein) and P06747 (rabies virus phosphoprotein) is 25%, while the bootstrap value for node defining the clade containg of Lagos bat virus phosphoprotein (O56773) and Mokola virus phosphoprotein (P0C569) is 100%. The bootstrap values for each of these clades is the percent of 100 bootstrap trees that the clade appears in.</p>
<p>Therefore, we are very confident that Lagos bat virus and Mokola virus phosphoproteins should be grouped together in the tree. However, we are not so confident that the Western Caucasian bat virus and rabies virus phosphoproteins should be grouped together.</p>
<p>The lengths of the branches in the plot of the tree are proportional to the amount of evolutionary change (estimated number of mutations) along the branches.</p>
<p>In this case, the branches leading to Lagos bat virus phosphoprotein (O56773) and Mokola virus phosphoprotein (P0C569) from the node representing their common ancestor are slightly shorter than the branches leading to the Western Caucasian bat virus (Q5VKP1) and rabies virus (P06747) phosphoproteins from the node representing their common ancestor.</p>
<p>This suggests that there might have been more mutations in the Western Caucasian bat virus (Q5VKP1) and rabies virus (P06747) phosphoproteins since they shared a common ancestor, than in the Lagos bat virus phosphoprotein (O56773) and Mokola virus phosphoprotein (P0C569) since they shared a common ancestor.</p>
<p>The tree above of the virus phosphoproteins is an unrooted phylogenetic tree as it does not contain an outgroup sequence, that is a sequence of a protein that is known to be more distantly related to the other proteins in the tree than they are to each other.</p>
<p>As a result, we cannot tell which direction evolutionary time ran in along the internal branches of the tree. For example, we cannot tell whether the node representing the common ancestor of (O56773, P0C569) was an ancestor of the node representing the common ancestor of (Q5VKP1, P06747), or the other way around.</p>
<p>In order to build a rooted phylogenetic tree, we need to have an outgroup sequence in our tree. In the case of the virus phosphoproteins, this is unfortunately not possible, as (as far as I know) there is not any protein known that is more distantly related to the four proteins already in our tree than they are to each other.</p>
<p>However, in many other cases, an outgroup - a sequence known to be more distantly relatd to the other sequences in the tree than they are to each other - is known, and so it is possible to build a rooted phylogenetic tree.</p>
<p>We discussed above that it is a good idea to investigate whether discarding the poorly conserved regions of a multiple alignment has an effect on the phylogenetic analysis. In this case, we made a filtered copy of the multiple alignment and stored it in the variable cleanedvirusaln (see above). We can make a phylogenetic tree based this filtered alignment, and see if it agrees with the phylogenetic tree based on the original alignment:</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#cb99-1"></a>cleanedvirusalntree &lt;-<span class="st"> </span><span class="kw">unrootedNJtree</span>(cleanedvirusaln,<span class="dt">type=</span><span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<p>Here O56773 and P0C569 are grouped together, and Q5VKP1 and P06747 are grouped together, as in the phylogenetic tree based on the raw (unfiltered) multiple alignment (see above). Thus, filtering the multiple alignment does not have an effect on the tree.</p>
<p>If we had found a difference in the trees made using the unfiltered and filtered multiple alignments, we would have to examine the multiple alignments closely, to see if the unfiltered multiple alignment contains a lot of very poorly aligned regions that might be adding noise to the phylogenetic analysis (if this is true, the tree based on the filtered alignment is likely to be more reliable).</p>

</div>
<div id="testing-the-significance-of-an-alignment" class="section level2">
<h2><span class="header-section-number">14.2</span> Testing the significance of an alignment</h2>
<p><strong>By</strong>: Avril Coghlan. Adapted and edited by Nathan Brouwer.</p>
</div>
<div id="calculating-the-statistical-significance-of-a-pairwise-global-alignment" class="section level2">
<h2><span class="header-section-number">14.3</span> Calculating the statistical significance of a pairwise global alignment</h2>
<p>We have seen that when we align the ‘PAWHEAE’ and ‘HEAGAWGHEE’ protein sequences, they have some similarity, and the score for their optimal global alignment is -5.</p>
<p>But is this alignment statistically significant? In other words, is this alignment better than we would expect between any two random proteins?</p>
<p>The Needleman-Wunsch alignment algorithm will produce a global alignment even if we give it two unrelated random protein sequences, although the alignment score would be low.</p>
<p>Therefore, we should ask: is the score for our alignment better than expected between two random sequences of the same lengths and amino acid compositions?</p>
<p>It is reasonable to expect that if the alignment score is statistically significant, then it will be higher than the scores obtained from aligning pairs of random protein sequences that have the same lengths and amino acid compositions as our original two sequences.</p>
<p>Therefore, to assess if the score for our alignment between the “PAWHEAE” and “HEAGAWGHEE” protein sequence is statistically significant, a first step is to make some random sequences that have the same amino acid composition and length as one of our initial two sequences, for example, as the same amino acid composition and length as the sequence ‘PAWHEAE’.</p>
<p>How can we obtain random sequences of the same amino acid composition and length as the sequence ‘PAWHEAE’? One way is to generate sequences using a multinomial model for protein sequences in which the probabilities of the different amino acids set to be equal to their frequencies in the sequence ‘PAWHEAE’.</p>
<p>That is, we can generate sequences using a multinomial model for proteins, in which the probability of “P” is set to 0.1428571 (1/7); the probability of “A” is set to 0.2857143 (2/7); the probability of “W” is set to 0.1428571 (1/7); the probability of "H’ is set to 0.1428571 (1/7); and the probabilty of ‘E’ is set to 0.2857143 (2/7), and the probabilities of the other 15 amino acids are set to 0.</p>
<p>To generate a sequence with this multinomial model, we choose the letter for each position in the sequence according to those probabilities. This is as if we have made a roulette wheel in which 1/7<em>th</em> of the circle is taken up by a pie labelled “P”, 2/7<em>ths</em> by a pie labelled “A”, 1/7<em>th</em> by a pie labelled “W”, 1/7<em>th</em> by a pie labelled “H”, and 2/7<em>ths</em> by a pie labelled “E”:</p>
<p>image3</p>
<p>To generate a sequence using the multinomial model, we keep spinning the arrow in the centre of the roulette wheel, and write down the letter that the arrow stops on after each spin. To generate a sequence that is 7 letters long, we can spin the arrow 7 times. To generate 1000 sequences that are each 7 letters long, we can spin the arrow 7000 times, where the letters chosen form 1000 7-letter amino acid sequences.</p>
<p>To generate a certain number (eg.1000) random amino acid sequences of a certain length using a multinomial model, you can use the function generateSeqsWithMultinomialModel() below:</p>
<p>The function generateSeqsWithMultinomialModel() generates X random sequences with a multinomial model, where the probabilities of the different letters are set equal to their frequencies in an input sequence, which is passed to the function as a string of characters (eg. ‘PAWHEAE’).</p>
<p>The function returns X random sequences in the form of a vector which has X elements, the first element of the vector contains the first sequence, the second element contains the second sequence, and so on.</p>
<p>You will need to copy and paste this function into R before you can use it.</p>
<p>We can use this function to generate 1000 7-letter amino acid sequences using a multinomial model in which the probabilities of the letters are set equal to their frequencies in ‘PAWHEAE’ (ie. probabilities 1/7 for P, 2/7 for A, 1/7 for W, 1/7 for H and 2/7 for E), by typing:</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#cb100-1"></a>randomseqs &lt;-<span class="st"> </span><span class="kw">generateSeqsWithMultinomialModel</span>(<span class="st">&#39;PAWHEAE&#39;</span>,<span class="dv">1000</span>)</span>
<span id="cb100-2"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#cb100-2"></a>randomseqs[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>] <span class="co"># Print out the first 10 random sequences</span></span></code></pre></div>
<p>The 1000 random sequences are stored in a vector randomseqs that has 1000 elements, each of which contains one of the random sequences.</p>
<p>We can then use the Needleman-Wunsch algorithm to align the sequence ‘HEAGAWGHEE’ to one of the 1000 random sequences generated using the multinomial model with probabilities 1/7 for P, 2/7 for A, 1/7 for W, 1/7 for H and 2/7 for E.</p>
<p>For example, to align ‘HEAGAWGHEE’ to the first of the 1000 random sequences (‘EEHAAAE’), we type:</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#cb101-1"></a>s4 &lt;-<span class="st"> &quot;HEAGAWGHEE&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#cb102-1"></a><span class="kw">pairwiseAlignment</span>(s4, randomseqs[<span class="dv">1</span>], <span class="dt">substitutionMatrix =</span> <span class="st">&quot;BLOSUM50&quot;</span>, <span class="dt">gapOpening =</span> <span class="dv">-2</span>,</span>
<span id="cb102-2"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#cb102-2"></a>  <span class="dt">gapExtension =</span> <span class="dv">-8</span>, <span class="dt">scoreOnly =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<p>If we use the pairwiseAlignment() function with the argument ‘scoreOnly=TRUE’, it will just give us the score for the alignment:</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#cb103-1"></a><span class="kw">pairwiseAlignment</span>(s4, randomseqs[<span class="dv">1</span>], <span class="dt">substitutionMatrix =</span> <span class="st">&quot;BLOSUM50&quot;</span>, <span class="dt">gapOpening =</span> <span class="dv">-2</span>,</span>
<span id="cb103-2"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#cb103-2"></a>  <span class="dt">gapExtension =</span> <span class="dv">-8</span>, <span class="dt">scoreOnly =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p>If we repeat this 1000 times, that is, for each of the 1000 random sequences in vector randomseqs, we can get a distribution of alignment scores expected for aligning ‘HEAGAWGHEE’ to random sequences of the same length and (approximately the same) amino acid composition as ‘PAWHEAE’.</p>
<p>We can then compare the actual score for aligning ‘PAWHEAE’ to ‘HEAGAWGHEE’ (ie. -5) to the distribution of scores for aligning ‘HEAGAWGHEE’ to the random sequences.</p>
<p>The code above first uses the double() function to create a numeric vector randomscores for storing real numbers (ie. not integers), with 1000 elements. This will be used to store the alignment scores for 1000 alignments between ‘HEAGAWGHEE’ and the 1000 different random sequences generated using the multinomial model.</p>
<p>The ‘for loop’ takes each of the 1000 different random sequences, aligns each one to ‘HEAGAWGHEE’, and stores the 1000 alignment scores in the randomscores vector.</p>
<p>Once we have run the ‘for loop’, we can make a histogram plot of the 1000 scores in vector randomscores by typing:</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#cb104-1"></a><span class="kw">hist</span>(randomscores, <span class="dt">col=</span><span class="st">&quot;red&quot;</span>) <span class="co"># Draw a red histogram</span></span></code></pre></div>
<p>image4</p>
<p>We can see from the histogram that quite a lot of the random sequences seem to have higher alignment scores than -5 when aligned to ‘HEAGAWGHEE’ (where -5 is the alignment score for ‘PAWHEAE’ and ‘HEAGAWGHEE’).</p>
<p>We can use the vector randomscores of scores for 1000 alignments of random sequences to ‘HEAGAWGHEE’ to calculate the probability of getting a score as large as the real alignment score for ‘PAWHEAE’ and ‘HEAGAWGHEE’ (ie. -5) by chance.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="unrooted-neighbor-joining-phylogenetic-trees.html#cb105-1"></a><span class="kw">sum</span>(randomscores <span class="op">&gt;=</span><span class="st"> </span><span class="dv">-5</span>)</span></code></pre></div>
<p>We see that 266 of the 1000 alignments of random sequences to ‘HEAGAWGHEE’ had alignment scores that were equal to or greater than -5. Thus, we can estimate that the probability of getting a score as large as the real alignment score by chance is (266/1000 =) 0.266. In other words, we can calculate a P-value of 0.266. This probability or P-value is quite high (almost 30%, or 1 in 3), so we can conclude that it is quite probable that we could get an alignment score as high as -5 by chance alone. This indicates that the sequences ‘HEAGAWGHEE’ and ‘PAWHEAE’ are not more similar than any two random sequences, and so they are probably not related sequences.</p>
<p>Another way of saying this is that the P-value that we calculated is high (0.266), and as a result we conclude that the alignment score for the sequences ‘HEAGAWGHEE’ and ‘PAWHEAE’ is not statistically significant. Generally, if the P-value that we calculate for an alignment of two sequences is &gt;0.05, we conclude that the alignment score is not statistically significant, and that the sequences are probably not related. On the other hand, if the P-value is less than or equal to 0.05, we conclude that the alignment score is statistically significant, and the sequences are very probably related (homologous).</p>
</div>
<div id="summary-1" class="section level2">
<h2><span class="header-section-number">14.4</span> Summary</h2>
<p>In this practical, you will have learned to use the following R functions:</p>
<p>data() for reading in data that comes with an R package
double() for creating a numeric vector for storing real (non-integer) numbers
toupper() for converting a string of characters from lowercase to uppercase
All of these functions belong to the standard installation of R.</p>
<p>You have also learned the following R functions that belong to the bioinformatics packages:</p>
<p>nucleotideSubstitutionMatrix() in the Biostrings package for making a nucleotide scoring matrix
pairwiseAlignment() in the Biostrings package for making a global alignment between two sequences
c2s() in the SeqinR package for converting a sequence stored in a vector to a string of characters
Links and Further Reading
Some links are included here for further reading.</p>
<p>For background reading on sequence alignment, it is recommended to read Chapter 3 of Introduction to Computational Genomics: a case studies approach by Cristianini and Hahn (Cambridge University Press; www.computational-genomics.net/book/).</p>
<p>For more in-depth information and more examples on using the SeqinR package for sequence analysis, look at the SeqinR documentation, <a href="http://pbil.univ-lyon1.fr/software/seqinr/doc.php?lang=eng" class="uri">http://pbil.univ-lyon1.fr/software/seqinr/doc.php?lang=eng</a>.</p>
<p>There is also a very nice chapter on “Analyzing Sequences”, which includes examples of using SeqinR and Biostrings for sequence analysis, as well as details on how to implement algorithms such as Needleman-Wunsch and Smith-Waterman in R yourself, in the book Applied statistics for bioinformatics using R by Krijnen (available online at cran.r-project.org/doc/contrib/Krijnen-IntroBioInfStatistics.pdf).</p>
<p>For a more in-depth introduction to R, a good online tutorial is available on the “Kickstarting R” website, cran.r-project.org/doc/contrib/Lemon-kickstart.</p>
<p>There is another nice (slightly more in-depth) tutorial to R available on the “Introduction to R” website, cran.r-project.org/doc/manuals/R-intro.html.</p>
<p>For more information on and examples using the Biostrings package, see the Biostrings documentation at <a href="http://www.bioconductor.org/packages/release/bioc/html/Biostrings.html" class="uri">http://www.bioconductor.org/packages/release/bioc/html/Biostrings.html</a>.</p>
<p>Acknowledgements
Many of the ideas for the examples and exercises for this practical were inspired by the Matlab case study on the Eyeless protein (www.computational-genomics.net/case_studies/eyeless_demo.html) from the website that accompanies the book Introduction to Computational Genomics: a case studies approach by Cristianini and Hahn (Cambridge University Press; www.computational-genomics.net/book/).</p>
<p>The examples of DNA sequences and protein sequences to align (‘GAATTC’ and ‘GATTA’, and sequences ‘PAWHEAE’ and ‘HEAGAWGHEE’), as well as some ideas related to finding the statistical significance of a pairwise alignment, were inspired by the chapter on “Analyzing Sequences” in the book Applied statistics for bioinformatics using R by Krijnen (cran.r-project.org/doc/contrib/Krijnen-IntroBioInfStatistics.pdf).</p>
<p>Thank you to Jean Lobry and Simon Penel for helpful advice on using the SeqinR package.</p>
<p>Contact
I will be grateful if you will send me (Avril Coghlan) corrections or suggestions for improvements to my email address <a href="mailto:alc@sanger.ac.uk" class="email">alc@sanger.ac.uk</a></p>
<p>License
The content in this book is licensed under a Creative Commons Attribution 3.0 License.</p>
<p>Exercises
Answer the following questions, using the R package. For each question, please record your answer, and what you typed into R to get this answer.</p>
<p>Model answers to the exercises are given in Answers to the exercises on Sequence Alignment.</p>
<ol style="list-style-type: decimal">
<li>Download FASTA-format files of the Brugia malayi Vab-3 protein (UniProt accession A8PZ80) and the Loa loa Vab-3 protein (UniProt accession E1FTG0) sequences from UniProt.
Note: the vab-3 gene of Brugia malayi and the vab-3 gene of Loa loa are related genes that control eye development in these two species. Brugia malayi and Loa loa are both parasitic nematode worms, which both cause filariasis, which is classified by the WHO as a neglected tropical disease.</li>
<li>What is the alignment score for the optimal global alignment between the Brugia malayi Vab-3 protein and the Loa loa Vab-3 protein, when you use the BLOSUM50 scoring matrix, a gap opening penalty of -10 and a gap extension penalty of -0.5?
Note: to specify a gap opening penalty of -10 and a gap extension penalty of -0.5, set the “gapOpening” argument to -9.5, and the “gapExtension” penalty to -0.5 in the pairwiseAlignment() function.</li>
<li>Use the printPairwiseAlignment() function to view the optimal global alignment between <em>Brugia malayi</em> Vab-3 protein and the Loa loa Vab-3 protein, using the BLOSUM50 scoring matrix, a gap opening penalty of -10 and a gap extension penalty of -0.5.
Do you see any regions where the alignment is very good (lots of identities and few gaps)?</li>
<li>What global alignment score do you get for the two Vab-3 proteins, when you use the BLOSUM62 alignment matrix, a gap opening penalty of -10 and a gap extension penalty of -0.5?
Which scoring matrix do you think is more appropriate for using for this pair of proteins: BLOSUM50 or BLOSUM62?</li>
<li>What is the statistical significance of the optimal global alignment for the Brugia malayi and Loa loa Vab-3 proteins made using the BLOSUM50 scoring matrix, with a gap opening penalty of -10 and a gap extension penalty of -0.5?
In other words, what is the probability of getting a score as large as the real alignment score for Vab-3 by chance?</li>
<li>What is the optimal global alignment score between the Brugia malayi Vab-6 protein and the Mycobacterium leprae chorismate lyase protein?
Is the alignment score statistically significant (what is the P- value?)? Does this surprise you?</li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="multiple-alignment-with-clustal.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="retrieving-multiple-sequences-in-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["compbio4all.pdf", "compbio4all.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
